<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 DNA Sequence Analysis | LSAMP Bioinformatics Short Course at Roxbury Community College</title>
  <meta name="description" content="7 DNA Sequence Analysis | LSAMP Bioinformatics Short Course at Roxbury Community College" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="7 DNA Sequence Analysis | LSAMP Bioinformatics Short Course at Roxbury Community College" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="/cover.jpg" />
  
  <meta name="github-repo" content="nikolaussuchert/bioinformatics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 DNA Sequence Analysis | LSAMP Bioinformatics Short Course at Roxbury Community College" />
  <meta name="twitter:site" content="@NikolausSucher" />
  
  <meta name="twitter:image" content="/cover.jpg" />




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="techniques-of-molecular-biology.html"/>
<link rel="next" href="pcr-primer-design.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="css/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><strong><a href="./">LSAMP Bioinformatics Worshop at RCC</a></strong></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="1" data-path="course-description.html"><a href="course-description.html"><i class="fa fa-check"></i><b>1</b> Course Description</a><ul>
<li class="chapter" data-level="1.1" data-path="course-description.html"><a href="course-description.html#prerequisites"><i class="fa fa-check"></i><b>1.1</b> Prerequisites</a></li>
<li class="chapter" data-level="1.2" data-path="course-description.html"><a href="course-description.html#instructional-objectives"><i class="fa fa-check"></i><b>1.2</b> Instructional Objectives:</a></li>
<li class="chapter" data-level="1.3" data-path="course-description.html"><a href="course-description.html#student-learning-outcomes"><i class="fa fa-check"></i><b>1.3</b> Student Learning Outcomes:</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="bioinformatics.html"><a href="bioinformatics.html"><i class="fa fa-check"></i><b>2</b> Bioinformatics</a><ul>
<li class="chapter" data-level="2.1" data-path="bioinformatics.html"><a href="bioinformatics.html#sequence-analysis"><i class="fa fa-check"></i><b>2.1</b> Sequence analysis</a><ul>
<li class="chapter" data-level="2.1.1" data-path="bioinformatics.html"><a href="bioinformatics.html#dna-sequencing"><i class="fa fa-check"></i><b>2.1.1</b> DNA sequencing</a></li>
<li class="chapter" data-level="2.1.2" data-path="bioinformatics.html"><a href="bioinformatics.html#sequence-assembly"><i class="fa fa-check"></i><b>2.1.2</b> Sequence assembly</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="bioinformatics.html"><a href="bioinformatics.html#genome-annotation"><i class="fa fa-check"></i><b>2.2</b> Genome annotation</a></li>
<li class="chapter" data-level="2.3" data-path="bioinformatics.html"><a href="bioinformatics.html#computational-evolutionary-biology"><i class="fa fa-check"></i><b>2.3</b> Computational evolutionary biology</a></li>
<li class="chapter" data-level="2.4" data-path="bioinformatics.html"><a href="bioinformatics.html#comparative-genomics"><i class="fa fa-check"></i><b>2.4</b> Comparative genomics</a></li>
<li class="chapter" data-level="2.5" data-path="bioinformatics.html"><a href="bioinformatics.html#pan-genomics"><i class="fa fa-check"></i><b>2.5</b> Pan genomics</a></li>
<li class="chapter" data-level="2.6" data-path="bioinformatics.html"><a href="bioinformatics.html#genetics-of-disease"><i class="fa fa-check"></i><b>2.6</b> Genetics of disease</a></li>
<li class="chapter" data-level="2.7" data-path="bioinformatics.html"><a href="bioinformatics.html#analysis-of-mutations-in-cancer"><i class="fa fa-check"></i><b>2.7</b> Analysis of mutations in cancer</a></li>
<li class="chapter" data-level="2.8" data-path="bioinformatics.html"><a href="bioinformatics.html#gene-and-protein-expression"><i class="fa fa-check"></i><b>2.8</b> Gene and protein expression</a><ul>
<li class="chapter" data-level="2.8.1" data-path="bioinformatics.html"><a href="bioinformatics.html#analysis-of-gene-expression"><i class="fa fa-check"></i><b>2.8.1</b> Analysis of gene expression</a></li>
<li class="chapter" data-level="2.8.2" data-path="bioinformatics.html"><a href="bioinformatics.html#analysis-of-protein-expression"><i class="fa fa-check"></i><b>2.8.2</b> Analysis of protein expression</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="bioinformatics.html"><a href="bioinformatics.html#analysis-of-regulation"><i class="fa fa-check"></i><b>2.9</b> Analysis of regulation</a></li>
<li class="chapter" data-level="2.10" data-path="bioinformatics.html"><a href="bioinformatics.html#analysis-of-cellular-organization"><i class="fa fa-check"></i><b>2.10</b> Analysis of cellular organization</a></li>
<li class="chapter" data-level="2.11" data-path="bioinformatics.html"><a href="bioinformatics.html#microscopy-and-image-analysis"><i class="fa fa-check"></i><b>2.11</b> Microscopy and image analysis</a><ul>
<li class="chapter" data-level="2.11.1" data-path="bioinformatics.html"><a href="bioinformatics.html#protein-localization"><i class="fa fa-check"></i><b>2.11.1</b> Protein localization</a></li>
</ul></li>
<li class="chapter" data-level="2.12" data-path="bioinformatics.html"><a href="bioinformatics.html#nuclear-organization-of-chromatin"><i class="fa fa-check"></i><b>2.12</b> Nuclear organization of chromatin</a></li>
<li class="chapter" data-level="2.13" data-path="bioinformatics.html"><a href="bioinformatics.html#structural-bioinformatics"><i class="fa fa-check"></i><b>2.13</b> Structural bioinformatics</a></li>
<li class="chapter" data-level="2.14" data-path="bioinformatics.html"><a href="bioinformatics.html#network-and-systems-biology"><i class="fa fa-check"></i><b>2.14</b> Network and systems biology</a></li>
<li class="chapter" data-level="2.15" data-path="bioinformatics.html"><a href="bioinformatics.html#literature-analysis"><i class="fa fa-check"></i><b>2.15</b> Literature analysis</a></li>
<li class="chapter" data-level="2.16" data-path="bioinformatics.html"><a href="bioinformatics.html#high-throughput-image-analysis"><i class="fa fa-check"></i><b>2.16</b> High-throughput image analysis</a></li>
<li class="chapter" data-level="2.17" data-path="bioinformatics.html"><a href="bioinformatics.html#high-throughput-single-cell-data-analysis"><i class="fa fa-check"></i><b>2.17</b> High-throughput single cell data analysis</a></li>
<li class="chapter" data-level="2.18" data-path="bioinformatics.html"><a href="bioinformatics.html#biodiversity-informatics"><i class="fa fa-check"></i><b>2.18</b> Biodiversity informatics</a></li>
<li class="chapter" data-level="2.19" data-path="bioinformatics.html"><a href="bioinformatics.html#ontologies-and-data-integration"><i class="fa fa-check"></i><b>2.19</b> Ontologies and data integration</a></li>
<li class="chapter" data-level="2.20" data-path="bioinformatics.html"><a href="bioinformatics.html#databases"><i class="fa fa-check"></i><b>2.20</b> Databases</a></li>
<li class="chapter" data-level="2.21" data-path="bioinformatics.html"><a href="bioinformatics.html#open-source-bioinformatics-software"><i class="fa fa-check"></i><b>2.21</b> Open-source bioinformatics software</a></li>
<li class="chapter" data-level="2.22" data-path="bioinformatics.html"><a href="bioinformatics.html#web-services-in-bioinformatics"><i class="fa fa-check"></i><b>2.22</b> Web services in bioinformatics</a></li>
<li class="chapter" data-level="2.23" data-path="bioinformatics.html"><a href="bioinformatics.html#bioinformatics-workflow-management-systems"><i class="fa fa-check"></i><b>2.23</b> Bioinformatics workflow management systems</a></li>
<li class="chapter" data-level="2.24" data-path="bioinformatics.html"><a href="bioinformatics.html#biocompute-and-biocompute-objects"><i class="fa fa-check"></i><b>2.24</b> BioCompute and BioCompute Objects</a></li>
<li class="chapter" data-level="2.25" data-path="bioinformatics.html"><a href="bioinformatics.html#education-platforms"><i class="fa fa-check"></i><b>2.25</b> Education platforms</a></li>
<li class="chapter" data-level="2.26" data-path="bioinformatics.html"><a href="bioinformatics.html#conferences"><i class="fa fa-check"></i><b>2.26</b> Conferences</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="dna-and-rna.html"><a href="dna-and-rna.html"><i class="fa fa-check"></i><b>3</b> DNA and RNA</a><ul>
<li class="chapter" data-level="3.1" data-path="dna-and-rna.html"><a href="dna-and-rna.html#dna"><i class="fa fa-check"></i><b>3.1</b> DNA</a><ul>
<li class="chapter" data-level="3.1.1" data-path="dna-and-rna.html"><a href="dna-and-rna.html#structure-and-composition-of-dna"><i class="fa fa-check"></i><b>3.1.1</b> Structure and composition of DNA</a></li>
<li class="chapter" data-level="3.1.2" data-path="dna-and-rna.html"><a href="dna-and-rna.html#base-modifications-and-dna-packaging"><i class="fa fa-check"></i><b>3.1.2</b> Base modifications and DNA packaging</a></li>
<li class="chapter" data-level="3.1.3" data-path="dna-and-rna.html"><a href="dna-and-rna.html#dna-damage"><i class="fa fa-check"></i><b>3.1.3</b> DNA damage</a></li>
<li class="chapter" data-level="3.1.4" data-path="dna-and-rna.html"><a href="dna-and-rna.html#biological-functions-of-dna"><i class="fa fa-check"></i><b>3.1.4</b> Biological functions of DNA</a></li>
<li class="chapter" data-level="3.1.5" data-path="dna-and-rna.html"><a href="dna-and-rna.html#dna-replication"><i class="fa fa-check"></i><b>3.1.5</b> DNA replication</a></li>
<li class="chapter" data-level="3.1.6" data-path="dna-and-rna.html"><a href="dna-and-rna.html#interactions-of-dna-with-proteins"><i class="fa fa-check"></i><b>3.1.6</b> Interactions of DNA with proteins</a></li>
<li class="chapter" data-level="3.1.7" data-path="dna-and-rna.html"><a href="dna-and-rna.html#dna-modifying-enzymes"><i class="fa fa-check"></i><b>3.1.7</b> DNA-modifying enzymes</a></li>
<li class="chapter" data-level="3.1.8" data-path="dna-and-rna.html"><a href="dna-and-rna.html#dna-recombination"><i class="fa fa-check"></i><b>3.1.8</b> DNA recombination</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="dna-and-rna.html"><a href="dna-and-rna.html#evolution"><i class="fa fa-check"></i><b>3.2</b> Evolution</a></li>
<li class="chapter" data-level="3.3" data-path="dna-and-rna.html"><a href="dna-and-rna.html#genetic-engineering"><i class="fa fa-check"></i><b>3.3</b> Genetic engineering</a></li>
<li class="chapter" data-level="3.4" data-path="dna-and-rna.html"><a href="dna-and-rna.html#dna-profiling"><i class="fa fa-check"></i><b>3.4</b> DNA profiling</a></li>
<li class="chapter" data-level="3.5" data-path="dna-and-rna.html"><a href="dna-and-rna.html#rna"><i class="fa fa-check"></i><b>3.5</b> RNA</a><ul>
<li class="chapter" data-level="3.5.1" data-path="dna-and-rna.html"><a href="dna-and-rna.html#comparison-with-dna"><i class="fa fa-check"></i><b>3.5.1</b> Comparison with DNA</a></li>
<li class="chapter" data-level="3.5.2" data-path="dna-and-rna.html"><a href="dna-and-rna.html#structure-of-rna"><i class="fa fa-check"></i><b>3.5.2</b> Structure of RNA</a></li>
<li class="chapter" data-level="3.5.3" data-path="dna-and-rna.html"><a href="dna-and-rna.html#synthesis-of-rna"><i class="fa fa-check"></i><b>3.5.3</b> Synthesis of RNA</a></li>
<li class="chapter" data-level="3.5.4" data-path="dna-and-rna.html"><a href="dna-and-rna.html#coding-and-non-coding-rna"><i class="fa fa-check"></i><b>3.5.4</b> Coding and non-coding RNA</a></li>
<li class="chapter" data-level="3.5.5" data-path="dna-and-rna.html"><a href="dna-and-rna.html#regulatory-rna"><i class="fa fa-check"></i><b>3.5.5</b> Regulatory RNA</a></li>
<li class="chapter" data-level="3.5.6" data-path="dna-and-rna.html"><a href="dna-and-rna.html#rna-interference-by-mirnas"><i class="fa fa-check"></i><b>3.5.6</b> RNA interference by miRNAs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html"><i class="fa fa-check"></i><b>4</b> Transcription And RNA Processing</a><ul>
<li class="chapter" data-level="4.1" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#initiation"><i class="fa fa-check"></i><b>4.1</b> Initiation</a></li>
<li class="chapter" data-level="4.2" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#elongation"><i class="fa fa-check"></i><b>4.2</b> Elongation</a></li>
<li class="chapter" data-level="4.3" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#termination"><i class="fa fa-check"></i><b>4.3</b> Termination</a></li>
<li class="chapter" data-level="4.4" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#inhibitors-of-transcription"><i class="fa fa-check"></i><b>4.4</b> Inhibitors of transcription</a></li>
<li class="chapter" data-level="4.5" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#transcription-factors"><i class="fa fa-check"></i><b>4.5</b> Transcription factors</a></li>
<li class="chapter" data-level="4.6" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#rna-processing"><i class="fa fa-check"></i><b>4.6</b> RNA Processing</a></li>
<li class="chapter" data-level="4.7" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#mrna-processing"><i class="fa fa-check"></i><b>4.7</b> mRNA processing</a><ul>
<li class="chapter" data-level="4.7.1" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#processing"><i class="fa fa-check"></i><b>4.7.1</b> 5’ processing</a></li>
<li class="chapter" data-level="4.7.2" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#processing-1"><i class="fa fa-check"></i><b>4.7.2</b> 3’ processing</a></li>
<li class="chapter" data-level="4.7.3" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#intron-splicing"><i class="fa fa-check"></i><b>4.7.3</b> Intron splicing</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#reverse-transcription"><i class="fa fa-check"></i><b>4.8</b> Reverse transcription</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html"><i class="fa fa-check"></i><b>5</b> Translation And The Genetic Code</a><ul>
<li class="chapter" data-level="5.1" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#ribosomes"><i class="fa fa-check"></i><b>5.1</b> Ribosomes</a></li>
<li class="chapter" data-level="5.2" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#bacterial-ribosomes"><i class="fa fa-check"></i><b>5.2</b> Bacterial ribosomes</a></li>
<li class="chapter" data-level="5.3" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#eukaryotic-ribosomes"><i class="fa fa-check"></i><b>5.3</b> Eukaryotic ribosomes</a></li>
<li class="chapter" data-level="5.4" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#translation"><i class="fa fa-check"></i><b>5.4</b> Translation</a></li>
<li class="chapter" data-level="5.5" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#eukaryotic-translation"><i class="fa fa-check"></i><b>5.5</b> Eukaryotic translation</a><ul>
<li class="chapter" data-level="5.5.1" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#initiation-1"><i class="fa fa-check"></i><b>5.5.1</b> Initiation</a></li>
<li class="chapter" data-level="5.5.2" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#elongation-1"><i class="fa fa-check"></i><b>5.5.2</b> Elongation</a></li>
<li class="chapter" data-level="5.5.3" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#termination-1"><i class="fa fa-check"></i><b>5.5.3</b> Termination</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#genetic-code"><i class="fa fa-check"></i><b>5.6</b> Genetic code</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html"><i class="fa fa-check"></i><b>6</b> Techniques Of Molecular Biology</a><ul>
<li class="chapter" data-level="6.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#polymerase-chain-reaction"><i class="fa fa-check"></i><b>6.1</b> Polymerase chain reaction</a><ul>
<li class="chapter" data-level="6.1.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#procedure"><i class="fa fa-check"></i><b>6.1.1</b> Procedure</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#dna-sequencing-1"><i class="fa fa-check"></i><b>6.2</b> DNA Sequencing</a><ul>
<li class="chapter" data-level="6.2.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#sanger-sequencing"><i class="fa fa-check"></i><b>6.2.1</b> Sanger sequencing</a></li>
<li class="chapter" data-level="6.2.2" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#high-throughput-sequencing-hts-methods"><i class="fa fa-check"></i><b>6.2.2</b> High-throughput sequencing (HTS) methods</a></li>
<li class="chapter" data-level="6.2.3" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#pyrosequencing"><i class="fa fa-check"></i><b>6.2.3</b> 454 pyrosequencing</a></li>
<li class="chapter" data-level="6.2.4" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#illumina-solexa-sequencing"><i class="fa fa-check"></i><b>6.2.4</b> Illumina (Solexa) sequencing</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#rna-seq"><i class="fa fa-check"></i><b>6.3</b> RNA-Seq</a><ul>
<li class="chapter" data-level="6.3.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#single-cell-rna-sequencing-scrna-seq"><i class="fa fa-check"></i><b>6.3.1</b> Single-cell RNA sequencing (scRNA-Seq)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#dna-microarray"><i class="fa fa-check"></i><b>6.4</b> DNA microarray</a></li>
<li class="chapter" data-level="6.5" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#recombinant-dna"><i class="fa fa-check"></i><b>6.5</b> Recombinant DNA</a></li>
<li class="chapter" data-level="6.6" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#molecular-cloning"><i class="fa fa-check"></i><b>6.6</b> Molecular cloning</a><ul>
<li class="chapter" data-level="6.6.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#procedures"><i class="fa fa-check"></i><b>6.6.1</b> Procedures</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#mutagenesis"><i class="fa fa-check"></i><b>6.7</b> Mutagenesis</a><ul>
<li class="chapter" data-level="6.7.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#random-mutagenesis"><i class="fa fa-check"></i><b>6.7.1</b> Random mutagenesis</a></li>
<li class="chapter" data-level="6.7.2" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#site-directed-mutagenesis"><i class="fa fa-check"></i><b>6.7.2</b> Site-directed mutagenesis</a></li>
<li class="chapter" data-level="6.7.3" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#gene-synthesis"><i class="fa fa-check"></i><b>6.7.3</b> Gene synthesis</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#gene-knockout"><i class="fa fa-check"></i><b>6.8</b> Gene knockout</a></li>
<li class="chapter" data-level="6.9" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#conditional-gene-knockout"><i class="fa fa-check"></i><b>6.9</b> Conditional gene knockout</a></li>
<li class="chapter" data-level="6.10" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#gene-knock-in"><i class="fa fa-check"></i><b>6.10</b> Gene knock-in</a></li>
<li class="chapter" data-level="6.11" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#rna-interference"><i class="fa fa-check"></i><b>6.11</b> RNA interference</a><ul>
<li class="chapter" data-level="6.11.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#transcriptional-silencing"><i class="fa fa-check"></i><b>6.11.1</b> Transcriptional silencing</a></li>
<li class="chapter" data-level="6.11.2" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#gene-knockdown"><i class="fa fa-check"></i><b>6.11.2</b> Gene knockdown</a></li>
</ul></li>
<li class="chapter" data-level="6.12" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#cre-lox-recombination"><i class="fa fa-check"></i><b>6.12</b> Cre-Lox recombination</a></li>
<li class="chapter" data-level="6.13" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#crispr-gene-editing"><i class="fa fa-check"></i><b>6.13</b> CRISPR gene editing</a></li>
<li class="chapter" data-level="6.14" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#gene-drive"><i class="fa fa-check"></i><b>6.14</b> Gene drive</a></li>
<li class="chapter" data-level="6.15" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#human-germline-modification"><i class="fa fa-check"></i><b>6.15</b> Human germline modification</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="dna-sequence-analysis.html"><a href="dna-sequence-analysis.html"><i class="fa fa-check"></i><b>7</b> DNA Sequence Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="dna-sequence-analysis.html"><a href="dna-sequence-analysis.html#dna-sequence-files"><i class="fa fa-check"></i><b>7.1</b> DNA Sequence Files</a></li>
<li class="chapter" data-level="7.2" data-path="dna-sequence-analysis.html"><a href="dna-sequence-analysis.html#blast-search"><i class="fa fa-check"></i><b>7.2</b> BLAST Search</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="pcr-primer-design.html"><a href="pcr-primer-design.html"><i class="fa fa-check"></i><b>8</b> PCR Primer Design</a></li>
<li class="chapter" data-level="9" data-path="the-human-genome.html"><a href="the-human-genome.html"><i class="fa fa-check"></i><b>9</b> The Human Genome</a><ul>
<li class="chapter" data-level="9.1" data-path="the-human-genome.html"><a href="the-human-genome.html#the-human-genome-project"><i class="fa fa-check"></i><b>9.1</b> The Human Genome Project</a></li>
<li class="chapter" data-level="9.2" data-path="the-human-genome.html"><a href="the-human-genome.html#public-private-competition-around-the-human-genome-project"><i class="fa fa-check"></i><b>9.2</b> Public-Private Competition Around The Human Genome Project</a></li>
<li class="chapter" data-level="9.3" data-path="the-human-genome.html"><a href="the-human-genome.html#genome-annotation-1"><i class="fa fa-check"></i><b>9.3</b> Genome Annotation</a></li>
<li class="chapter" data-level="9.4" data-path="the-human-genome.html"><a href="the-human-genome.html#international-hapmap-project"><i class="fa fa-check"></i><b>9.4</b> International HapMap Project</a></li>
<li class="chapter" data-level="9.5" data-path="the-human-genome.html"><a href="the-human-genome.html#the-single-nucleotide-polymorphism-database-dbsnp-of-nucleotide-sequence-variation"><i class="fa fa-check"></i><b>9.5</b> The Single Nucleotide Polymorphism Database (dbSNP) of Nucleotide Sequence Variation</a></li>
<li class="chapter" data-level="9.6" data-path="the-human-genome.html"><a href="the-human-genome.html#the-1000-genomes-project"><i class="fa fa-check"></i><b>9.6</b> The 1000 Genomes Project</a></li>
<li class="chapter" data-level="9.7" data-path="the-human-genome.html"><a href="the-human-genome.html#personal-genomics"><i class="fa fa-check"></i><b>9.7</b> Personal genomics</a></li>
<li class="chapter" data-level="9.8" data-path="the-human-genome.html"><a href="the-human-genome.html#exploring-the-human-genome"><i class="fa fa-check"></i><b>9.8</b> Exploring The Human Genome</a><ul>
<li class="chapter" data-level="9.8.1" data-path="the-human-genome.html"><a href="the-human-genome.html#experimental-procedures"><i class="fa fa-check"></i><b>9.8.1</b> Experimental Procedures</a></li>
</ul></li>
<li class="chapter" data-level="9.9" data-path="the-human-genome.html"><a href="the-human-genome.html#review-questions"><i class="fa fa-check"></i><b>9.9</b> Review Questions</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="r-rstudio-cloud-and-bioconductor.html"><a href="r-rstudio-cloud-and-bioconductor.html"><i class="fa fa-check"></i><b>10</b> R, RStudio Cloud, And BioConductor</a><ul>
<li class="chapter" data-level="10.1" data-path="r-rstudio-cloud-and-bioconductor.html"><a href="r-rstudio-cloud-and-bioconductor.html#an-overview-of-r-rstudio-and-bioconductor"><i class="fa fa-check"></i><b>10.1</b> An Overview of R, RStudio and BioConductor</a></li>
<li class="chapter" data-level="10.2" data-path="r-rstudio-cloud-and-bioconductor.html"><a href="r-rstudio-cloud-and-bioconductor.html#rstudio-cloud"><i class="fa fa-check"></i><b>10.2</b> RStudio Cloud</a></li>
<li class="chapter" data-level="10.3" data-path="r-rstudio-cloud-and-bioconductor.html"><a href="r-rstudio-cloud-and-bioconductor.html#basic-syntax-of-r"><i class="fa fa-check"></i><b>10.3</b> Basic syntax of R</a></li>
<li class="chapter" data-level="10.4" data-path="r-rstudio-cloud-and-bioconductor.html"><a href="r-rstudio-cloud-and-bioconductor.html#install-bioconductor-and-bioconductor-packages"><i class="fa fa-check"></i><b>10.4</b> Install Bioconductor and Bioconductor Packages</a></li>
<li class="chapter" data-level="10.5" data-path="r-rstudio-cloud-and-bioconductor.html"><a href="r-rstudio-cloud-and-bioconductor.html#blast-from-r"><i class="fa fa-check"></i><b>10.5</b> BLAST From R</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">LSAMP Bioinformatics Short Course at Roxbury Community College</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dna-sequence-analysis" class="section level1">
<h1><span class="header-section-number">7</span> DNA Sequence Analysis</h1>
<p>In this exercise, we will learn how to work with DNA sequence data. We will learn how to use various online bioinformatics software tools. We will use a DNA sequence that was obtained in a previous experiment. In the experiment, genomic DNA was isolated from an organism and used as template in a polymerase change reaction (PCR). The primers for the PCR were chosen in the coding region of a gene that is commonly used for “barcoding” of biological species belonging to one of the four kingdoms of eykaryotic organisms. The PCR products were sent to a commercial DNA sequencing facility for sequencing by the Sanger method.</p>
<div id="dna-sequence-files" class="section level2">
<h2><span class="header-section-number">7.1</span> DNA Sequence Files</h2>
<p>The sequencing facility sent the results back in the form of two types of files for each sequenced sample: one type of file has the file extension <a href="https://projects.nfstc.org/workshops/resources/articles/ABIF_File_Format.pdf">.ab1</a>. Files of this type are binary files and contain DNA sequence information recorded by an <a href="https://en.wikipedia.org/wiki/Applied_Biosystems">Applied Biosystems</a> DNA sequencer and associated software; also known as a electropherogram file or DNA trace file; can be be viewed graphically by a ABI file viewer to analyze and compare DNA sequences. For conveniencem a second file with the file extension <em>.seq</em> or .txt contains the DNA sequence in <a href="https://en.wikipedia.org/wiki/ASCII">ASCII</a> (American Standard Code for Information Interchange) format. Files of this type can be read by any program that can read simple text.</p>
<p>The electropherogram of the sequence that we will be using in this exercise is shown in Figure <a href="dna-sequence-analysis.html#fig:chrom">7.1</a> below. An <a href="https://en.wikipedia.org/wiki/Electropherogram">electropherogram</a>, or electrophoregram, is a record or chart produced by an automated <a href="https://en.wikipedia.org/wiki/DNA_sequencer">DNA sequencer</a> used to separate the labled DNA produced in the Sanger sequencing procedure.</p>
<ol style="list-style-type: decimal">
<li>Inspect the electropherograme (electrophoregram) shown in Fig. <a href="dna-sequence-analysis.html#fig:chrom">7.1</a>.</li>
<li>Notice peaks of four different colors (green, blue, black and red) corresponding to the bases adenine (A), cytosine (C), guanine (G), and thymine (T) in the sequenced DNA. Narrow, well separated peaks indicate a good signal and high confidence in “calling the base”, broad, overlapping peaks signify low quality data. If the sofware algorithm used cannot decide which base to call, the letter “N” is used.</li>
<li>Notice how the peaks are broad and overlapping both at the very beginning and end of the sequence. This is typical.</li>
</ol>

<div class="figure" style="text-align: center"><span id="fig:chrom"></span>
<img src="data/sequence_chromatogram.pdf" alt="The DNA sequence chromatogram obtained from sequencing the PCR products from the unknown organism. This chromatogram was produced using the free and open-source software programs R and Bioconductor package sangerseqR." width="70%" />
<p class="caption">
Figure 7.1: The DNA sequence chromatogram obtained from sequencing the PCR products from the unknown organism. This chromatogram was produced using the free and open-source software programs <a href="https://www.r-project.org">R</a> and <a href="https://bioconductor.org">Bioconductor</a> package <a href="https://bioconductor.org/packages/release/bioc/html/sangerseqR.html">sangerseqR</a>.
</p>
</div>
</div>
<div id="blast-search" class="section level2">
<h2><span class="header-section-number">7.2</span> BLAST Search</h2>
<p>We will use the BLAST (Basic Local Alignment Tool) program to compare our DNA with all DNA sequences stored in GenBank. <a href="https://www.ncbi.nlm.nih.gov/genbank/">GenBank</a> is the NIH genetic sequence database, an annotated collection of all publicly available DNA sequences. GenBank is part of the international nucleotide sequence database collaboration, which comprises the DNA Databank of Japan (DDBJ), the European Nucleotide Archive (ENA), and GenBank at NCBI. these three organizations exchange data on a daily basis.</p>
<p>There are several ways to search and retrieve data from GenBank:</p>
<ol style="list-style-type: decimal">
<li>Search GenBank for sequence identifiers and annotations with entrez nucleotide, which is divided into three divisions:
<ul>
<li>corenucleotide (the main collection)</li>
<li>dbest (expressed sequence tags)</li>
<li>dbgss (genome survey sequences).</li>
</ul></li>
<li>Search and align GenBank sequences to a query sequence using BLAST (basic local alignment search tool).</li>
</ol>
<p>We will use BLAST to search te corenucleotide main collection.</p>
<ol style="list-style-type: decimal">
<li>Open a web browser on your computer. If you are reading this in your browser, right click on the highlighted link to open the web site of the <a href="https://www.nlm.nih.gov">U.S. National Library of Medicine</a> in a new tab (fig. <a href="dna-sequence-analysis.html#fig:nlm">7.2</a>). Otherwise enter the link in your browser manually.</li>
</ol>

<div class="figure" style="text-align: center"><span id="fig:nlm"></span>
<img src="figures/bioinformatics/nlm.png" alt="The web page of the U.S. National Library of Medicine." width="70%" />
<p class="caption">
Figure 7.2: The web page of the U.S. National Library of Medicine.
</p>
</div>
<ol start="2" style="list-style-type: decimal">
<li>On the U.S. National Library of Medicine web page (fig. <a href="dna-sequence-analysis.html#fig:nlm">7.2</a>), click on the right most square with the DNA icon and BLAST (basic local alignment tool) written on it.</li>
<li>On the newly opened page (fig. <a href="dna-sequence-analysis.html#fig:blast">7.3</a>), you will see a row of squares displaying various titles. Click on the left most square that has nucleotide BLAST written on it. A new page will open.</li>
</ol>

<div class="figure" style="text-align: center"><span id="fig:blast"></span>
<img src="figures/bioinformatics/blast.png" alt="The Basic Local Alignment Search Tool (BLAST) start page." width="70%" />
<p class="caption">
Figure 7.3: The Basic Local Alignment Search Tool (BLAST) start page.
</p>
</div>
<ol start="4" style="list-style-type: decimal">
<li>Here is the text only version of DNA electropherogram shown in Figure <a href="dna-sequence-analysis.html#fig:chrom">7.1</a> above. Copy the sequence to your clipboard:</li>
</ol>
<pre><code>NNNNNNNNNNNNNNNNNNNNNNNNNNNTAANNNANNCTTCTTTTTTACATTTAACAAGATTTTTTNTACATGAATATAAAGAAA
GGTCTAAAAATGAAAAATTGATTTATTCTTTTAATAAAAGAAATAAATTTGTCACATTTTCTTGGAATTTTTTTTATATTTTTG
AATTGGAATTTTTTTTAACTTCTCTTTTGACTAGGTTTNTAAATAATTTAGTTCTATCCAACTTATTCGATCAAATTAATCTAT
TAGAAAAAATAAATAATAATAATAAAAGTCAATATTTTTTATCAGAAAAAAGGATCTATAACCAAAATTCCTGTATTCATTATG
TACGGTATCAAAATCGTTGTATTATGGCTTCGGAAGGCTTTTATTTTCATGATACGAACTGGATATATTTTATATTGAATATTT
GGCAATTTTTTATGCATTTATGGATTCAACCTTTCAGGTTCTCAACAAAGCATTTTCAAAAGCAGAGTTTTTTTTTTCTGGGTT
ATCAATTTGGCAGAGAATCCAAACTGCTAAAAGTTCGATCTATTTCACTTGATAAATCACCTACAATTTATTCACGTTTAAAAA
AAAATCTTTTAAAAACACAAATTGTATATCCAATAGATTTTCTAGCAAAAGAGGGTTTTTGTGATATTTCAGGTTATCCTATTA
GTAGATCGACTTGGACTACNTCGACNGATGAAGAANTTTTATTGAATTTTAATAAAATTTGNAAAAGCTTTTATTTTTATTATG
GGGGGTTGATAAAAAAAGACATTTTATANNNAATCANNTATATCCTCCGATTTTCATGTGCAAAAACNCTANCTCGTAANCNNN
NAANNNNNAANNNNNNA</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>Paste the copied DNA sequence into the white box on the top left (Fig. <a href="dna-sequence-analysis.html#fig:sequence">7.4</a>).</li>
</ol>

<div class="figure" style="text-align: center"><span id="fig:sequence"></span>
<img src="figures/bioinformatics/sequence_entry.png" alt="Nucleotide BLAST (BLASTN) sequence entry form." width="70%" />
<p class="caption">
Figure 7.4: Nucleotide BLAST (BLASTN) sequence entry form.
</p>
</div>
<ol start="6" style="list-style-type: decimal">
<li>Leave the default values unchanged and click on the oval blue button at the lower left that has BLAST written on it (Fig. <a href="dna-sequence-analysis.html#fig:click">7.5</a>). This will upload your sequence (referred to from now on as the <strong>Query</strong> sequence) to the server where it will be compared to all sequences on record.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:click"></span>
<img src="figures/bioinformatics/click_blast.png" alt="After you have pasted your sequence into the sequence entry field, click the BLAST button at the lower left of the page." width="70%" />
<p class="caption">
Figure 7.5: After you have pasted your sequence into the sequence entry field, click the BLAST button at the lower left of the page.
</p>
</div>
<ol start="7" style="list-style-type: decimal">
<li>A new page will open that will be updated every 2 seconds while your query is being processed (Fig. <a href="dna-sequence-analysis.html#fig:status">7.6</a>).</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:status"></span>
<img src="figures/bioinformatics/job_id.png" alt="The BLASTN query status updated page." width="70%" />
<p class="caption">
Figure 7.6: The BLASTN query status updated page.
</p>
</div>
<ol start="8" style="list-style-type: decimal">
<li>Once the search has completed, the BLASTN results page will open (Fig. <a href="dna-sequence-analysis.html#fig:results">7.7</a>).</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:results"></span>
<img src="figures/bioinformatics/results.png" alt="The BLASTN results page." width="70%" />
<p class="caption">
Figure 7.7: The BLASTN results page.
</p>
</div>
<ol start="9" style="list-style-type: decimal">
<li>Click on the “+” sign in front of “Graphic Summary” (Fig. <a href="dna-sequence-analysis.html#fig:results">7.7</a>). A graphical summary of your results will be shown (Fig. <a href="dna-sequence-analysis.html#fig:graphic">7.8</a>. The light green line on top represents your query sequence, below are shown any retreived sequences that align with your sequence.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:graphic"></span>
<img src="figures/bioinformatics/graphic_summary.png" alt="Graphic summary of the BLASTN search results." width="70%" />
<p class="caption">
Figure 7.8: Graphic summary of the BLASTN search results.
</p>
</div>
<ol start="10" style="list-style-type: decimal">
<li>Click on the “+” sign in front of “Description” (Fig. <a href="dna-sequence-analysis.html#fig:description">7.9</a>). A list of sequences that align with your query sequence will be shown (Fig. <a href="dna-sequence-analysis.html#fig:description">7.9</a>.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:description"></span>
<img src="figures/bioinformatics/results_description.png" alt="List of sequences producing significant alignments with your query sequence." width="70%" />
<p class="caption">
Figure 7.9: List of sequences producing significant alignments with your query sequence.
</p>
</div>
<ol start="11" style="list-style-type: decimal">
<li>Click on the first sequence. A new page will showing the alignment of the query sequence with the retrieved sequence (Fig. <a href="dna-sequence-analysis.html#fig:alignment">7.10</a>).</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:alignment"></span>
<img src="figures/bioinformatics/alignment.png" alt="Alignment of the best retrieved sequence with the query sequence." width="70%" />
<p class="caption">
Figure 7.10: Alignment of the best retrieved sequence with the query sequence.
</p>
</div>
<ol start="12" style="list-style-type: decimal">
<li><p>Which sequence matches our query sequence?</p></li>
<li><p>Is this sequence linked to a scientific publication? If so, download the original paper.</p></li>
<li><p>What is the scientific name of the organism that has a matching sequence? What is the common English name of this organism? What type of organism is it?</p></li>
<li><p>What is the name of the gene that matches this sequence? Does the query sequence correspond to a partial or the full-length sequence of this gene?</p></li>
<li><p>Go back to the list of sequence alignments produced by BLAST. Look for the sequnce with GenBank accession AY348551.1 (accession identifiers are listed in the left most column). Click on it. A new page will open. 15. On which chromosome is this gene located? How many nucleotides does this chromosome have?</p></li>
<li><p>How many amino acids does the full-lenght gene code for?</p></li>
<li><p>Find open reading frame (ORF) in the sequence. Write down the nucleotide mumber corresponding to the the first base of the start codon (<strong>A</strong>TG). Find the stop codon. Write down the number of the last nucleotide of the stop codon. You will need those numbers in the next exercise (“Design of PCR Primers”).</p></li>
<li><p>What is the function of this gene in the organism?</p></li>
<li><p>Click on the “GenBank” link on the top left of the page and choose “FASTA”.</p></li>
</ol>

<div class="figure" style="text-align: center"><span id="fig:fastap"></span>
<img src="figures/bioinformatics/NCBI_FASTA_view.png" alt="View of the DNA seuence with GenBank accession # AY348551.1 in FASTA format." width="70%" />
<p class="caption">
Figure 7.11: View of the DNA seuence with GenBank accession # AY348551.1 in FASTA format.
</p>
</div>
<p>In bioinformatics and biochemistry, the <a href="https://en.wikipedia.org/wiki/FASTA_format">FASTA format</a> is a text-based format for representing either nucleotide sequences or amino acid (protein) sequences, in which nucleotides or amino acids are represented using single-letter codes. The format also allows for sequence names and comments to precede the sequences. The format originates from the <a href="https://en.wikipedia.org/wiki/FASTA">FASTA software package</a>, but has now become a near universal standard in the field of bioinformatics.</p>
<p>In the original format, a sequence was represented as a series of lines, each of which was no longer than 120 characters and usually did not exceed 80 characters. This probably was to allow for preallocation of fixed line sizes in software: at the time most users relied on Digital Equipment Corporation (DEC) VT220 (or compatible) terminals which could display 80 or 132 characters per line.[citation needed] Most people preferred the bigger font in 80-character modes and so it became the recommended fashion to use 80 characters or less (often 70) in FASTA lines. Also, the width of a standard printed page is 70 to 80 characters (depending on the font). Hence, 80 characters became the norm.</p>
<p>The first line in a FASTA file started either with a “&gt;” (greater-than) symbol or, less frequently, a “;” (semicolon) was taken as a comment. Subsequent lines starting with a semicolon would be ignored by software. Since the only comment used was the first, it quickly became used to hold a summary description of the sequence, often starting with a unique library accession number, and with time it has become commonplace to always use “&gt;” for the first line and to not use “;” comments (which would otherwise be ignored).</p>
<p>Following the initial line (used for a unique description of the sequence) was the actual sequence itself in standard one-letter character string. Anything other than a valid character would be ignored (including spaces, tabulators, asterisks, etc…). It was also common to end the sequence with an "*" (asterisk) character (in analogy with use in PIR formatted sequences) and, for the same reason, to leave a blank line between the description and the sequence.</p>
<ol start="21" style="list-style-type: decimal">
<li>Towards the top left of the page, just above the title, click on “Send to” (Fig. <a href="dna-sequence-analysis.html#fig:fasta">7.12</a>), in the pop-up, make sure that “Complete Record” is highlighted, and under “Choose Destination”, click on “File”, then under “Format”, choose “Fasta”, and then click on “Create File”. Save the file to your computer.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:fasta"></span>
<img src="figures/bioinformatics/save-fasta.png" alt="Save the gene sequence in FASTA format" width="70%" />
<p class="caption">
Figure 7.12: Save the gene sequence in FASTA format
</p>
</div>
<ol start="22" style="list-style-type: decimal">
<li>Click on the “GenBank” link on the top left of the page and choose “Graphics”. Click on the different buttons and explore the many options and tools available. For example, show the open reading frames in all six frames. Find the open reading frame for the matK protein.</li>
</ol>

<div class="figure" style="text-align: center"><span id="fig:graphicsview"></span>
<img src="figures/bioinformatics/NCBI_graphics_view.png" alt="Graphics view of the DNA seuence with GenBank accession # AY348551.1." width="70%" />
<p class="caption">
Figure 7.13: Graphics view of the DNA seuence with GenBank accession # AY348551.1.
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="techniques-of-molecular-biology.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="pcr-primer-design.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": false,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/nikolaussucher/bioinformatics/edit/master/sequence_analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
