<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>10 R, RStudio Cloud, And BioConductor | LSAMP Bioinformatics Short Course at Roxbury Community College</title>
  <meta name="description" content="10 R, RStudio Cloud, And BioConductor | LSAMP Bioinformatics Short Course at Roxbury Community College" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="10 R, RStudio Cloud, And BioConductor | LSAMP Bioinformatics Short Course at Roxbury Community College" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="/cover.jpg" />
  
  <meta name="github-repo" content="nikolaussuchert/bioinformatics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="10 R, RStudio Cloud, And BioConductor | LSAMP Bioinformatics Short Course at Roxbury Community College" />
  <meta name="twitter:site" content="@NikolausSucher" />
  
  <meta name="twitter:image" content="/cover.jpg" />




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="the-human-genome.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="css/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><strong><a href="./">LSAMP Bioinformatics Worshop at RCC</a></strong></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="1" data-path="course-description.html"><a href="course-description.html"><i class="fa fa-check"></i><b>1</b> Course Description</a><ul>
<li class="chapter" data-level="1.1" data-path="course-description.html"><a href="course-description.html#prerequisites"><i class="fa fa-check"></i><b>1.1</b> Prerequisites</a></li>
<li class="chapter" data-level="1.2" data-path="course-description.html"><a href="course-description.html#instructional-objectives"><i class="fa fa-check"></i><b>1.2</b> Instructional Objectives:</a></li>
<li class="chapter" data-level="1.3" data-path="course-description.html"><a href="course-description.html#student-learning-outcomes"><i class="fa fa-check"></i><b>1.3</b> Student Learning Outcomes:</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="bioinformatics.html"><a href="bioinformatics.html"><i class="fa fa-check"></i><b>2</b> Bioinformatics</a><ul>
<li class="chapter" data-level="2.1" data-path="bioinformatics.html"><a href="bioinformatics.html#sequence-analysis"><i class="fa fa-check"></i><b>2.1</b> Sequence analysis</a><ul>
<li class="chapter" data-level="2.1.1" data-path="bioinformatics.html"><a href="bioinformatics.html#dna-sequencing"><i class="fa fa-check"></i><b>2.1.1</b> DNA sequencing</a></li>
<li class="chapter" data-level="2.1.2" data-path="bioinformatics.html"><a href="bioinformatics.html#sequence-assembly"><i class="fa fa-check"></i><b>2.1.2</b> Sequence assembly</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="bioinformatics.html"><a href="bioinformatics.html#genome-annotation"><i class="fa fa-check"></i><b>2.2</b> Genome annotation</a></li>
<li class="chapter" data-level="2.3" data-path="bioinformatics.html"><a href="bioinformatics.html#computational-evolutionary-biology"><i class="fa fa-check"></i><b>2.3</b> Computational evolutionary biology</a></li>
<li class="chapter" data-level="2.4" data-path="bioinformatics.html"><a href="bioinformatics.html#comparative-genomics"><i class="fa fa-check"></i><b>2.4</b> Comparative genomics</a></li>
<li class="chapter" data-level="2.5" data-path="bioinformatics.html"><a href="bioinformatics.html#pan-genomics"><i class="fa fa-check"></i><b>2.5</b> Pan genomics</a></li>
<li class="chapter" data-level="2.6" data-path="bioinformatics.html"><a href="bioinformatics.html#genetics-of-disease"><i class="fa fa-check"></i><b>2.6</b> Genetics of disease</a></li>
<li class="chapter" data-level="2.7" data-path="bioinformatics.html"><a href="bioinformatics.html#analysis-of-mutations-in-cancer"><i class="fa fa-check"></i><b>2.7</b> Analysis of mutations in cancer</a></li>
<li class="chapter" data-level="2.8" data-path="bioinformatics.html"><a href="bioinformatics.html#gene-and-protein-expression"><i class="fa fa-check"></i><b>2.8</b> Gene and protein expression</a><ul>
<li class="chapter" data-level="2.8.1" data-path="bioinformatics.html"><a href="bioinformatics.html#analysis-of-gene-expression"><i class="fa fa-check"></i><b>2.8.1</b> Analysis of gene expression</a></li>
<li class="chapter" data-level="2.8.2" data-path="bioinformatics.html"><a href="bioinformatics.html#analysis-of-protein-expression"><i class="fa fa-check"></i><b>2.8.2</b> Analysis of protein expression</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="bioinformatics.html"><a href="bioinformatics.html#analysis-of-regulation"><i class="fa fa-check"></i><b>2.9</b> Analysis of regulation</a></li>
<li class="chapter" data-level="2.10" data-path="bioinformatics.html"><a href="bioinformatics.html#analysis-of-cellular-organization"><i class="fa fa-check"></i><b>2.10</b> Analysis of cellular organization</a></li>
<li class="chapter" data-level="2.11" data-path="bioinformatics.html"><a href="bioinformatics.html#microscopy-and-image-analysis"><i class="fa fa-check"></i><b>2.11</b> Microscopy and image analysis</a><ul>
<li class="chapter" data-level="2.11.1" data-path="bioinformatics.html"><a href="bioinformatics.html#protein-localization"><i class="fa fa-check"></i><b>2.11.1</b> Protein localization</a></li>
</ul></li>
<li class="chapter" data-level="2.12" data-path="bioinformatics.html"><a href="bioinformatics.html#nuclear-organization-of-chromatin"><i class="fa fa-check"></i><b>2.12</b> Nuclear organization of chromatin</a></li>
<li class="chapter" data-level="2.13" data-path="bioinformatics.html"><a href="bioinformatics.html#structural-bioinformatics"><i class="fa fa-check"></i><b>2.13</b> Structural bioinformatics</a></li>
<li class="chapter" data-level="2.14" data-path="bioinformatics.html"><a href="bioinformatics.html#network-and-systems-biology"><i class="fa fa-check"></i><b>2.14</b> Network and systems biology</a></li>
<li class="chapter" data-level="2.15" data-path="bioinformatics.html"><a href="bioinformatics.html#literature-analysis"><i class="fa fa-check"></i><b>2.15</b> Literature analysis</a></li>
<li class="chapter" data-level="2.16" data-path="bioinformatics.html"><a href="bioinformatics.html#high-throughput-image-analysis"><i class="fa fa-check"></i><b>2.16</b> High-throughput image analysis</a></li>
<li class="chapter" data-level="2.17" data-path="bioinformatics.html"><a href="bioinformatics.html#high-throughput-single-cell-data-analysis"><i class="fa fa-check"></i><b>2.17</b> High-throughput single cell data analysis</a></li>
<li class="chapter" data-level="2.18" data-path="bioinformatics.html"><a href="bioinformatics.html#biodiversity-informatics"><i class="fa fa-check"></i><b>2.18</b> Biodiversity informatics</a></li>
<li class="chapter" data-level="2.19" data-path="bioinformatics.html"><a href="bioinformatics.html#ontologies-and-data-integration"><i class="fa fa-check"></i><b>2.19</b> Ontologies and data integration</a></li>
<li class="chapter" data-level="2.20" data-path="bioinformatics.html"><a href="bioinformatics.html#databases"><i class="fa fa-check"></i><b>2.20</b> Databases</a></li>
<li class="chapter" data-level="2.21" data-path="bioinformatics.html"><a href="bioinformatics.html#open-source-bioinformatics-software"><i class="fa fa-check"></i><b>2.21</b> Open-source bioinformatics software</a></li>
<li class="chapter" data-level="2.22" data-path="bioinformatics.html"><a href="bioinformatics.html#web-services-in-bioinformatics"><i class="fa fa-check"></i><b>2.22</b> Web services in bioinformatics</a></li>
<li class="chapter" data-level="2.23" data-path="bioinformatics.html"><a href="bioinformatics.html#bioinformatics-workflow-management-systems"><i class="fa fa-check"></i><b>2.23</b> Bioinformatics workflow management systems</a></li>
<li class="chapter" data-level="2.24" data-path="bioinformatics.html"><a href="bioinformatics.html#biocompute-and-biocompute-objects"><i class="fa fa-check"></i><b>2.24</b> BioCompute and BioCompute Objects</a></li>
<li class="chapter" data-level="2.25" data-path="bioinformatics.html"><a href="bioinformatics.html#education-platforms"><i class="fa fa-check"></i><b>2.25</b> Education platforms</a></li>
<li class="chapter" data-level="2.26" data-path="bioinformatics.html"><a href="bioinformatics.html#conferences"><i class="fa fa-check"></i><b>2.26</b> Conferences</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="dna-and-rna.html"><a href="dna-and-rna.html"><i class="fa fa-check"></i><b>3</b> DNA and RNA</a><ul>
<li class="chapter" data-level="3.1" data-path="dna-and-rna.html"><a href="dna-and-rna.html#dna"><i class="fa fa-check"></i><b>3.1</b> DNA</a><ul>
<li class="chapter" data-level="3.1.1" data-path="dna-and-rna.html"><a href="dna-and-rna.html#structure-and-composition-of-dna"><i class="fa fa-check"></i><b>3.1.1</b> Structure and composition of DNA</a></li>
<li class="chapter" data-level="3.1.2" data-path="dna-and-rna.html"><a href="dna-and-rna.html#base-modifications-and-dna-packaging"><i class="fa fa-check"></i><b>3.1.2</b> Base modifications and DNA packaging</a></li>
<li class="chapter" data-level="3.1.3" data-path="dna-and-rna.html"><a href="dna-and-rna.html#dna-damage"><i class="fa fa-check"></i><b>3.1.3</b> DNA damage</a></li>
<li class="chapter" data-level="3.1.4" data-path="dna-and-rna.html"><a href="dna-and-rna.html#biological-functions-of-dna"><i class="fa fa-check"></i><b>3.1.4</b> Biological functions of DNA</a></li>
<li class="chapter" data-level="3.1.5" data-path="dna-and-rna.html"><a href="dna-and-rna.html#dna-replication"><i class="fa fa-check"></i><b>3.1.5</b> DNA replication</a></li>
<li class="chapter" data-level="3.1.6" data-path="dna-and-rna.html"><a href="dna-and-rna.html#interactions-of-dna-with-proteins"><i class="fa fa-check"></i><b>3.1.6</b> Interactions of DNA with proteins</a></li>
<li class="chapter" data-level="3.1.7" data-path="dna-and-rna.html"><a href="dna-and-rna.html#dna-modifying-enzymes"><i class="fa fa-check"></i><b>3.1.7</b> DNA-modifying enzymes</a></li>
<li class="chapter" data-level="3.1.8" data-path="dna-and-rna.html"><a href="dna-and-rna.html#dna-recombination"><i class="fa fa-check"></i><b>3.1.8</b> DNA recombination</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="dna-and-rna.html"><a href="dna-and-rna.html#evolution"><i class="fa fa-check"></i><b>3.2</b> Evolution</a></li>
<li class="chapter" data-level="3.3" data-path="dna-and-rna.html"><a href="dna-and-rna.html#genetic-engineering"><i class="fa fa-check"></i><b>3.3</b> Genetic engineering</a></li>
<li class="chapter" data-level="3.4" data-path="dna-and-rna.html"><a href="dna-and-rna.html#dna-profiling"><i class="fa fa-check"></i><b>3.4</b> DNA profiling</a></li>
<li class="chapter" data-level="3.5" data-path="dna-and-rna.html"><a href="dna-and-rna.html#rna"><i class="fa fa-check"></i><b>3.5</b> RNA</a><ul>
<li class="chapter" data-level="3.5.1" data-path="dna-and-rna.html"><a href="dna-and-rna.html#comparison-with-dna"><i class="fa fa-check"></i><b>3.5.1</b> Comparison with DNA</a></li>
<li class="chapter" data-level="3.5.2" data-path="dna-and-rna.html"><a href="dna-and-rna.html#structure-of-rna"><i class="fa fa-check"></i><b>3.5.2</b> Structure of RNA</a></li>
<li class="chapter" data-level="3.5.3" data-path="dna-and-rna.html"><a href="dna-and-rna.html#synthesis-of-rna"><i class="fa fa-check"></i><b>3.5.3</b> Synthesis of RNA</a></li>
<li class="chapter" data-level="3.5.4" data-path="dna-and-rna.html"><a href="dna-and-rna.html#coding-and-non-coding-rna"><i class="fa fa-check"></i><b>3.5.4</b> Coding and non-coding RNA</a></li>
<li class="chapter" data-level="3.5.5" data-path="dna-and-rna.html"><a href="dna-and-rna.html#regulatory-rna"><i class="fa fa-check"></i><b>3.5.5</b> Regulatory RNA</a></li>
<li class="chapter" data-level="3.5.6" data-path="dna-and-rna.html"><a href="dna-and-rna.html#rna-interference-by-mirnas"><i class="fa fa-check"></i><b>3.5.6</b> RNA interference by miRNAs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html"><i class="fa fa-check"></i><b>4</b> Transcription And RNA Processing</a><ul>
<li class="chapter" data-level="4.1" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#initiation"><i class="fa fa-check"></i><b>4.1</b> Initiation</a></li>
<li class="chapter" data-level="4.2" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#elongation"><i class="fa fa-check"></i><b>4.2</b> Elongation</a></li>
<li class="chapter" data-level="4.3" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#termination"><i class="fa fa-check"></i><b>4.3</b> Termination</a></li>
<li class="chapter" data-level="4.4" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#inhibitors-of-transcription"><i class="fa fa-check"></i><b>4.4</b> Inhibitors of transcription</a></li>
<li class="chapter" data-level="4.5" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#transcription-factors"><i class="fa fa-check"></i><b>4.5</b> Transcription factors</a></li>
<li class="chapter" data-level="4.6" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#rna-processing"><i class="fa fa-check"></i><b>4.6</b> RNA Processing</a></li>
<li class="chapter" data-level="4.7" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#mrna-processing"><i class="fa fa-check"></i><b>4.7</b> mRNA processing</a><ul>
<li class="chapter" data-level="4.7.1" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#processing"><i class="fa fa-check"></i><b>4.7.1</b> 5’ processing</a></li>
<li class="chapter" data-level="4.7.2" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#processing-1"><i class="fa fa-check"></i><b>4.7.2</b> 3’ processing</a></li>
<li class="chapter" data-level="4.7.3" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#intron-splicing"><i class="fa fa-check"></i><b>4.7.3</b> Intron splicing</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="transcription-and-rna-processing.html"><a href="transcription-and-rna-processing.html#reverse-transcription"><i class="fa fa-check"></i><b>4.8</b> Reverse transcription</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html"><i class="fa fa-check"></i><b>5</b> Translation And The Genetic Code</a><ul>
<li class="chapter" data-level="5.1" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#ribosomes"><i class="fa fa-check"></i><b>5.1</b> Ribosomes</a></li>
<li class="chapter" data-level="5.2" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#bacterial-ribosomes"><i class="fa fa-check"></i><b>5.2</b> Bacterial ribosomes</a></li>
<li class="chapter" data-level="5.3" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#eukaryotic-ribosomes"><i class="fa fa-check"></i><b>5.3</b> Eukaryotic ribosomes</a></li>
<li class="chapter" data-level="5.4" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#translation"><i class="fa fa-check"></i><b>5.4</b> Translation</a></li>
<li class="chapter" data-level="5.5" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#eukaryotic-translation"><i class="fa fa-check"></i><b>5.5</b> Eukaryotic translation</a><ul>
<li class="chapter" data-level="5.5.1" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#initiation-1"><i class="fa fa-check"></i><b>5.5.1</b> Initiation</a></li>
<li class="chapter" data-level="5.5.2" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#elongation-1"><i class="fa fa-check"></i><b>5.5.2</b> Elongation</a></li>
<li class="chapter" data-level="5.5.3" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#termination-1"><i class="fa fa-check"></i><b>5.5.3</b> Termination</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="translation-and-the-genetic-code.html"><a href="translation-and-the-genetic-code.html#genetic-code"><i class="fa fa-check"></i><b>5.6</b> Genetic code</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html"><i class="fa fa-check"></i><b>6</b> Techniques Of Molecular Biology</a><ul>
<li class="chapter" data-level="6.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#polymerase-chain-reaction"><i class="fa fa-check"></i><b>6.1</b> Polymerase chain reaction</a><ul>
<li class="chapter" data-level="6.1.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#procedure"><i class="fa fa-check"></i><b>6.1.1</b> Procedure</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#dna-sequencing-1"><i class="fa fa-check"></i><b>6.2</b> DNA Sequencing</a><ul>
<li class="chapter" data-level="6.2.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#sanger-sequencing"><i class="fa fa-check"></i><b>6.2.1</b> Sanger sequencing</a></li>
<li class="chapter" data-level="6.2.2" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#high-throughput-sequencing-hts-methods"><i class="fa fa-check"></i><b>6.2.2</b> High-throughput sequencing (HTS) methods</a></li>
<li class="chapter" data-level="6.2.3" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#pyrosequencing"><i class="fa fa-check"></i><b>6.2.3</b> 454 pyrosequencing</a></li>
<li class="chapter" data-level="6.2.4" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#illumina-solexa-sequencing"><i class="fa fa-check"></i><b>6.2.4</b> Illumina (Solexa) sequencing</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#rna-seq"><i class="fa fa-check"></i><b>6.3</b> RNA-Seq</a><ul>
<li class="chapter" data-level="6.3.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#single-cell-rna-sequencing-scrna-seq"><i class="fa fa-check"></i><b>6.3.1</b> Single-cell RNA sequencing (scRNA-Seq)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#dna-microarray"><i class="fa fa-check"></i><b>6.4</b> DNA microarray</a></li>
<li class="chapter" data-level="6.5" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#recombinant-dna"><i class="fa fa-check"></i><b>6.5</b> Recombinant DNA</a></li>
<li class="chapter" data-level="6.6" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#molecular-cloning"><i class="fa fa-check"></i><b>6.6</b> Molecular cloning</a><ul>
<li class="chapter" data-level="6.6.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#procedures"><i class="fa fa-check"></i><b>6.6.1</b> Procedures</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#mutagenesis"><i class="fa fa-check"></i><b>6.7</b> Mutagenesis</a><ul>
<li class="chapter" data-level="6.7.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#random-mutagenesis"><i class="fa fa-check"></i><b>6.7.1</b> Random mutagenesis</a></li>
<li class="chapter" data-level="6.7.2" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#site-directed-mutagenesis"><i class="fa fa-check"></i><b>6.7.2</b> Site-directed mutagenesis</a></li>
<li class="chapter" data-level="6.7.3" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#gene-synthesis"><i class="fa fa-check"></i><b>6.7.3</b> Gene synthesis</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#gene-knockout"><i class="fa fa-check"></i><b>6.8</b> Gene knockout</a></li>
<li class="chapter" data-level="6.9" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#conditional-gene-knockout"><i class="fa fa-check"></i><b>6.9</b> Conditional gene knockout</a></li>
<li class="chapter" data-level="6.10" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#gene-knock-in"><i class="fa fa-check"></i><b>6.10</b> Gene knock-in</a></li>
<li class="chapter" data-level="6.11" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#rna-interference"><i class="fa fa-check"></i><b>6.11</b> RNA interference</a><ul>
<li class="chapter" data-level="6.11.1" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#transcriptional-silencing"><i class="fa fa-check"></i><b>6.11.1</b> Transcriptional silencing</a></li>
<li class="chapter" data-level="6.11.2" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#gene-knockdown"><i class="fa fa-check"></i><b>6.11.2</b> Gene knockdown</a></li>
</ul></li>
<li class="chapter" data-level="6.12" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#cre-lox-recombination"><i class="fa fa-check"></i><b>6.12</b> Cre-Lox recombination</a></li>
<li class="chapter" data-level="6.13" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#crispr-gene-editing"><i class="fa fa-check"></i><b>6.13</b> CRISPR gene editing</a></li>
<li class="chapter" data-level="6.14" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#gene-drive"><i class="fa fa-check"></i><b>6.14</b> Gene drive</a></li>
<li class="chapter" data-level="6.15" data-path="techniques-of-molecular-biology.html"><a href="techniques-of-molecular-biology.html#human-germline-modification"><i class="fa fa-check"></i><b>6.15</b> Human germline modification</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="dna-sequence-analysis.html"><a href="dna-sequence-analysis.html"><i class="fa fa-check"></i><b>7</b> DNA Sequence Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="dna-sequence-analysis.html"><a href="dna-sequence-analysis.html#dna-sequence-files"><i class="fa fa-check"></i><b>7.1</b> DNA Sequence Files</a></li>
<li class="chapter" data-level="7.2" data-path="dna-sequence-analysis.html"><a href="dna-sequence-analysis.html#blast-search"><i class="fa fa-check"></i><b>7.2</b> BLAST Search</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="pcr-primer-design.html"><a href="pcr-primer-design.html"><i class="fa fa-check"></i><b>8</b> PCR Primer Design</a></li>
<li class="chapter" data-level="9" data-path="the-human-genome.html"><a href="the-human-genome.html"><i class="fa fa-check"></i><b>9</b> The Human Genome</a><ul>
<li class="chapter" data-level="9.1" data-path="the-human-genome.html"><a href="the-human-genome.html#the-human-genome-project"><i class="fa fa-check"></i><b>9.1</b> The Human Genome Project</a></li>
<li class="chapter" data-level="9.2" data-path="the-human-genome.html"><a href="the-human-genome.html#genome-annotation-1"><i class="fa fa-check"></i><b>9.2</b> Genome Annotation</a></li>
<li class="chapter" data-level="9.3" data-path="the-human-genome.html"><a href="the-human-genome.html#single-nucleotide-polymorphism"><i class="fa fa-check"></i><b>9.3</b> Single-nucleotide Polymorphism</a></li>
<li class="chapter" data-level="9.4" data-path="the-human-genome.html"><a href="the-human-genome.html#the-single-nucleotide-polymorphism-database-dbsnp-of-nucleotide-sequence-variation"><i class="fa fa-check"></i><b>9.4</b> The Single Nucleotide Polymorphism Database (dbSNP) of Nucleotide Sequence Variation</a></li>
<li class="chapter" data-level="9.5" data-path="the-human-genome.html"><a href="the-human-genome.html#international-hapmap-project"><i class="fa fa-check"></i><b>9.5</b> International HapMap Project</a></li>
<li class="chapter" data-level="9.6" data-path="the-human-genome.html"><a href="the-human-genome.html#the-1000-genomes-project"><i class="fa fa-check"></i><b>9.6</b> The 1000 Genomes Project</a></li>
<li class="chapter" data-level="9.7" data-path="the-human-genome.html"><a href="the-human-genome.html#personal-genomics"><i class="fa fa-check"></i><b>9.7</b> Personal genomics</a></li>
<li class="chapter" data-level="9.8" data-path="the-human-genome.html"><a href="the-human-genome.html#exploring-the-human-genome"><i class="fa fa-check"></i><b>9.8</b> Exploring The Human Genome</a><ul>
<li class="chapter" data-level="9.8.1" data-path="the-human-genome.html"><a href="the-human-genome.html#experimental-procedures"><i class="fa fa-check"></i><b>9.8.1</b> Experimental Procedures</a></li>
</ul></li>
<li class="chapter" data-level="9.9" data-path="the-human-genome.html"><a href="the-human-genome.html#review-questions"><i class="fa fa-check"></i><b>9.9</b> Review Questions</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="r-rstudio-cloud-and-bioconductor.html"><a href="r-rstudio-cloud-and-bioconductor.html"><i class="fa fa-check"></i><b>10</b> R, RStudio Cloud, And BioConductor</a><ul>
<li class="chapter" data-level="10.1" data-path="r-rstudio-cloud-and-bioconductor.html"><a href="r-rstudio-cloud-and-bioconductor.html#an-overview-of-r-rstudio-and-bioconductor"><i class="fa fa-check"></i><b>10.1</b> An Overview of R, RStudio and BioConductor</a></li>
<li class="chapter" data-level="10.2" data-path="r-rstudio-cloud-and-bioconductor.html"><a href="r-rstudio-cloud-and-bioconductor.html#rstudio-cloud"><i class="fa fa-check"></i><b>10.2</b> RStudio Cloud</a></li>
<li class="chapter" data-level="10.3" data-path="r-rstudio-cloud-and-bioconductor.html"><a href="r-rstudio-cloud-and-bioconductor.html#basic-syntax-of-r"><i class="fa fa-check"></i><b>10.3</b> Basic syntax of R</a></li>
<li class="chapter" data-level="10.4" data-path="r-rstudio-cloud-and-bioconductor.html"><a href="r-rstudio-cloud-and-bioconductor.html#install-bioconductor-and-bioconductor-packages"><i class="fa fa-check"></i><b>10.4</b> Install Bioconductor and Bioconductor Packages</a></li>
<li class="chapter" data-level="10.5" data-path="r-rstudio-cloud-and-bioconductor.html"><a href="r-rstudio-cloud-and-bioconductor.html#blast-from-r"><i class="fa fa-check"></i><b>10.5</b> BLAST From R</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">LSAMP Bioinformatics Short Course at Roxbury Community College</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="r-rstudio-cloud-and-bioconductor" class="section level1">
<h1><span class="header-section-number">10</span> R, RStudio Cloud, And BioConductor</h1>
<p>In this exercise, we will learn how to use R, RStudio Cloud and BioConductor. While you can download R and RStudio for free and install the programs on your own computer, it is easier and convenient to use RStudio in the cloud. It is free for personal use. But let’s start with a quick overview first.</p>
<p><a href="https://www.r-project.org">R</a> is a programming language and free software environment for statistical computing and graphics supported by the R Foundation for Statistical Computing. The R language is widely used among statisticians and data miners for developing statistical software and data analysis. Polls, data mining surveys, and studies of scholarly literature databases show substantial increases in popularity; as of April 2021, R ranks 16th in the TIOBE index, a measure of popularity of programming languages.</p>
<p>RStudio is an Integrated Development Environment (IDE) for R, a programming language for statistical computing and graphics. It is available in two formats: RStudio Desktop is a regular desktop application while RStudio Server runs on a remote server and allows accessing RStudio using a web browser.</p>
<p>Bioconductor is a free, open source and open development software project for the analysis and comprehension of genomic data generated by wet lab experiments in molecular biology.</p>
<p>Bioconductor is based primarily on the statistical R programming language, but does contain contributions in other programming languages. It has two releases each year that follow the semiannual releases of R. At any one time there is a release version, which corresponds to the released version of R, and a development version, which corresponds to the development version of R. Most users will find the release version appropriate for their needs. In addition there are many genome annotation packages available that are mainly, but not solely, oriented towards different types of microarrays.</p>
<p>While computational methods continue to be developed to interpret biological data, the Bioconductor project is an open source software repository that hosts a wide range of statistical tools developed in the R programming environment. Utilizing a rich array of statistical and graphical features in R, many Bioconductor packages have been developed to meet various data analysis needs. The use of these packages provides a basic understanding of the R programming / command language. As a result, R and Bioconductor packages, which have a strong computing background, are used by most biologists who will benefit significantly from their ability to analyze datasets. All these results provide biologists with easy access to the analysis of genomic data without requiring programming expertise.</p>
<p>The project was started in the Fall of 2001 and is overseen by the Bioconductor core team, based primarily at the Fred Hutchinson Cancer Research Center, with other members coming from international institutions.</p>
<p>Most Bioconductor components are distributed as R packages, which are add-on modules for R. Initially most of the Bioconductor software packages focused on the analysis of single channel Affymetrix and two or more channel cDNA/Oligo microarrays. As the project has matured, the functional scope of the software packages broadened to include the analysis of all types of genomic data, such as SAGE, sequence, or SNP data.</p>
<div id="an-overview-of-r-rstudio-and-bioconductor" class="section level2">
<h2><span class="header-section-number">10.1</span> An Overview of R, RStudio and BioConductor</h2>
<p>R is an interpreted language; users typically access it through a command-line interpreter. If a user types 2+2 at the R command prompt and presses enter, the computer replies with 4, as shown below:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="r-rstudio-cloud-and-bioconductor.html#cb2-1"></a><span class="dv">2</span> <span class="op">+</span><span class="st"> </span><span class="dv">2</span></span>
<span id="cb2-2"><a href="r-rstudio-cloud-and-bioconductor.html#cb2-2"></a><span class="co">#&gt; [1] 4</span></span></code></pre></div>
<p>This calculation is interpreted as the sum of two single-element vectors, resulting in a single-element vector. The prefix indicates that the list of elements following it on the same line starts with the first element of the vector (a feature that is useful when the output extends over multiple lines).</p>
<p>Like other similar languages such as APL and MATLAB, R supports matrix arithmetic. R’s data structures include vectors, matrices, arrays, data frames (similar to tables in a relational database) and lists. Arrays are stored in colum-major order. R’s extensible object system includes objects for (among others): regression models, time-series and geo-spatial coordinates. The scalar data type was never a data structure of R. Instead, a scalar is represented as a vector with length one.</p>
<p>R supports procedural programming with functions and, for some functions, object-oriented programming with generic functions. A generic function acts differently depending on the classes of arguments passed to it. In other words, the generic function dispatches the function (method) specific to that class of object. For example, R has a generic print function that can print almost every class of object in R with a simple print(objectname) syntax.</p>
<p>The capabilities of R are extended through user-created packages, which allow specialised statistical techniques, graphical devices, import/export capabilities, reporting tools (RMarkdown, knitr, Sweave), etc. The large number of packages available for R, and the ease of installing and using them, has been cited as a major factor in driving the widespread adoption of the language in data science. The R packaging system is also used by researchers to create compendia to organise research data, code and report files in a systematic way for sharing and public archiving.</p>
<p>A core set of packages is included with the installation of R, with more than 15,000 additional packages (as of September 2018) available at the Comprehensive R Archive Network (CRAN), Bioconductor, Omegahat, GitHub, and other repositories.</p>
<p>The “Task Views” page (subject list) on the CRAN website lists a wide range of tasks (in fields such as Finance, Genetics, High Performance Computing, Machine Learning, Medical Imaging, Social Sciences and Spatial Statistics) to which R has been applied and for which packages are available. R has also been identified by the FDA as suitable for interpreting data from clinical research.</p>
<p><a href="https://www.bioconductor.org">The Bioconductor project</a> provides R packages for the analysis of genomic data. This includes object-oriented data-handling and analysis tools for data from Affymetrix, cDNA microarray, and next-generation high-throughput sequencing methods.</p>
<p>A group of packages called the Tidyverse, which can be considered a “dialect of the R language”, is increasingly popular in the R ecosystem. The group of packages strives to provide a cohesive collection of functions to deal with common data science tasks, including data import, cleaning, transformation and visualisation (notably with the ggplot2 package).</p>
<p>The official R software environment is a <a href="https://en.wikipedia.org/wiki/GNU_Project">GNU</a> package. It is written primarily in C, Fortran, and R itself (thus, it is partially self-hosting) and is freely available under the GNU General Public License. Pre-compiled executables are provided fr various operating systems. Although R has a command line interface, there are several third-party graphical user interfaces, such as <a href="https://www.rstudio.com">RStudio</a>, an integrated development environment, and Jupyter, a notebook interface.</p>
</div>
<div id="rstudio-cloud" class="section level2">
<h2><span class="header-section-number">10.2</span> RStudio Cloud</h2>
<p>RStudio Cloud allows anyone to do, share, teach and learn data science online. We will use it to learn how to use BioConductor.</p>
<p>There is nothing to configure and no dedicated hardware, installation or annual purchase contract required. We only need a browser.</p>
<p>Open a web browser and go to: <a href="https://rstudio.cloud" class="uri">https://rstudio.cloud</a></p>
<p>When the page opens, click on “GET STARTED FOR FREE”.</p>
<div class="figure" style="text-align: center"><span id="fig:rstudiocloud"></span>
<img src="figures/bioconductor/RStudio_cloud.png" alt="The RStudio Cloud landing page." width="70%" />
<p class="caption">
Figure 10.1: The RStudio Cloud landing page.
</p>
</div>
<p>On the next page, click on “Sign Up” on the bottom right.</p>
<div class="figure" style="text-align: center"><span id="fig:getstarted"></span>
<img src="figures/bioconductor/SignUp.png" alt="The RStudio Cloud &quot;Get started&quot; page." width="70%" />
<p class="caption">
Figure 10.2: The RStudio Cloud “Get started” page.
</p>
</div>
<p>Fill in the required information on the next page.</p>
<div class="figure" style="text-align: center"><span id="fig:signup"></span>
<img src="figures/bioconductor/SignUpForm.png" alt="The RStudio Cloud sign-up page." width="70%" />
<p class="caption">
Figure 10.3: The RStudio Cloud sign-up page.
</p>
</div>
<p>Once you are logged in, click on “New Project” on the top right. You can name the project anything you like, for example “Bioinformatics”.</p>
<div class="figure" style="text-align: center"><span id="fig:newproject"></span>
<img src="figures/bioconductor/NewProject.png" alt="Create a new poject." width="70%" />
<p class="caption">
Figure 10.4: Create a new poject.
</p>
</div>
<p>Then, click on the project. The RStudio workspace page will open.</p>
<div class="figure" style="text-align: center"><span id="fig:workspace"></span>
<img src="figures/bioconductor/Workspace.png" alt="The RStudio Workspace." width="70%" />
<p class="caption">
Figure 10.5: The RStudio Workspace.
</p>
</div>
</div>
<div id="basic-syntax-of-r" class="section level2">
<h2><span class="header-section-number">10.3</span> Basic syntax of R</h2>
<p>The following examples illustrate the basic syntax of the language and use of the command-line interface. (An expanded list of standard language features can be found in the R manual, “An Introduction to R”.)</p>
<p>In R, the generally preferred assignment operator is an arrow made from two characters &lt;-, although = can be used in some cases.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-1"></a></span>
<span id="cb3-2"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-2"></a>x &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="dv">6</span> <span class="co"># Create a numeric vector in the current environment</span></span>
<span id="cb3-3"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-3"></a>y &lt;-<span class="st"> </span>x<span class="op">^</span><span class="dv">2</span> <span class="co"># Create vector based on the values in x.</span></span>
<span id="cb3-4"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-4"></a><span class="kw">print</span>(y) <span class="co"># Print the vector’s contents.</span></span>
<span id="cb3-5"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-5"></a><span class="co">#&gt; [1]  1  4  9 16 25 36</span></span>
<span id="cb3-6"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-6"></a></span>
<span id="cb3-7"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-7"></a>z &lt;-<span class="st"> </span>x <span class="op">+</span><span class="st"> </span>y <span class="co"># Create a new vector that is the sum of x and y</span></span>
<span id="cb3-8"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-8"></a>z <span class="co"># return the contents of z to the current environment.</span></span>
<span id="cb3-9"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-9"></a><span class="co">#&gt; [1]  2  6 12 20 30 42</span></span>
<span id="cb3-10"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-10"></a></span>
<span id="cb3-11"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-11"></a>z_matrix &lt;-<span class="st"> </span><span class="kw">matrix</span>(z, <span class="dt">nrow=</span><span class="dv">3</span>) <span class="co"># Create a new matrix that tuns the vector z into a 3x2 matrix object</span></span>
<span id="cb3-12"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-12"></a>z_matrix </span>
<span id="cb3-13"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-13"></a><span class="co">#&gt;      [,1] [,2]</span></span>
<span id="cb3-14"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-14"></a><span class="co">#&gt; [1,]    2   20</span></span>
<span id="cb3-15"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-15"></a><span class="co">#&gt; [2,]    6   30</span></span>
<span id="cb3-16"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-16"></a><span class="co">#&gt; [3,]   12   42</span></span>
<span id="cb3-17"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-17"></a></span>
<span id="cb3-18"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-18"></a><span class="dv">2</span><span class="op">*</span><span class="kw">t</span>(z_matrix)<span class="op">-</span><span class="dv">2</span> <span class="co"># Transpose the matrix, multiply every element by 2, subtract 2 from each element in the matrix, and return the results to the terminal.</span></span>
<span id="cb3-19"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-19"></a><span class="co">#&gt;      [,1] [,2] [,3]</span></span>
<span id="cb3-20"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-20"></a><span class="co">#&gt; [1,]    2   10   22</span></span>
<span id="cb3-21"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-21"></a><span class="co">#&gt; [2,]   38   58   82</span></span>
<span id="cb3-22"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-22"></a></span>
<span id="cb3-23"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-23"></a>new_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">t</span>(z_matrix), <span class="dt">row.names=</span><span class="kw">c</span>(<span class="st">&#39;A&#39;</span>,<span class="st">&#39;B&#39;</span>)) <span class="co"># Create a new data.frame object that contains the data from a transposed z_matrix, with row names &#39;A&#39; and &#39;B&#39;</span></span>
<span id="cb3-24"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-24"></a></span>
<span id="cb3-25"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-25"></a><span class="kw">names</span>(new_df) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;X&#39;</span>,<span class="st">&#39;Y&#39;</span>,<span class="st">&#39;Z&#39;</span>) <span class="co"># set the column names of new_df as X, Y, and Z.</span></span>
<span id="cb3-26"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-26"></a></span>
<span id="cb3-27"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-27"></a><span class="kw">print</span>(new_df)  <span class="co">#print the current results.</span></span>
<span id="cb3-28"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-28"></a><span class="co">#&gt;    X  Y  Z</span></span>
<span id="cb3-29"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-29"></a><span class="co">#&gt; A  2  6 12</span></span>
<span id="cb3-30"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-30"></a><span class="co">#&gt; B 20 30 42</span></span>
<span id="cb3-31"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-31"></a></span>
<span id="cb3-32"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-32"></a>new_df<span class="op">$</span>Z <span class="co">#output the Z column</span></span>
<span id="cb3-33"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-33"></a><span class="co">#&gt; [1] 12 42</span></span>
<span id="cb3-34"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-34"></a></span>
<span id="cb3-35"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-35"></a>new_df<span class="op">$</span>Z<span class="op">==</span>new_df[<span class="st">&#39;Z&#39;</span>] <span class="op">&amp;&amp;</span><span class="st"> </span>new_df<span class="op">==</span>new_df<span class="op">$</span>Z <span class="co"># the data.frame column Z can be accessed using $Z, [&#39;Z&#39;], or  syntax, and the values are the same. </span></span>
<span id="cb3-36"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-36"></a><span class="co">#&gt; [1] FALSE</span></span>
<span id="cb3-37"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-37"></a></span>
<span id="cb3-38"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-38"></a><span class="kw">attributes</span>(new_df) <span class="co">#print attributes information about the new_df object</span></span>
<span id="cb3-39"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-39"></a><span class="co">#&gt; $names</span></span>
<span id="cb3-40"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-40"></a><span class="co">#&gt; [1] &quot;X&quot; &quot;Y&quot; &quot;Z&quot;</span></span>
<span id="cb3-41"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-41"></a><span class="co">#&gt; </span></span>
<span id="cb3-42"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-42"></a><span class="co">#&gt; $class</span></span>
<span id="cb3-43"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-43"></a><span class="co">#&gt; [1] &quot;data.frame&quot;</span></span>
<span id="cb3-44"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-44"></a><span class="co">#&gt; </span></span>
<span id="cb3-45"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-45"></a><span class="co">#&gt; $row.names</span></span>
<span id="cb3-46"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-46"></a><span class="co">#&gt; [1] &quot;A&quot; &quot;B&quot;</span></span>
<span id="cb3-47"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-47"></a></span>
<span id="cb3-48"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-48"></a><span class="kw">attributes</span>(new_df)<span class="op">$</span>row.names &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;one&#39;</span>,<span class="st">&#39;two&#39;</span>) <span class="co">## access and then change the row.names attribute; can also be done using rownames()</span></span>
<span id="cb3-49"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-49"></a>new_df</span>
<span id="cb3-50"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-50"></a><span class="co">#&gt;      X  Y  Z</span></span>
<span id="cb3-51"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-51"></a><span class="co">#&gt; one  2  6 12</span></span>
<span id="cb3-52"><a href="r-rstudio-cloud-and-bioconductor.html#cb3-52"></a><span class="co">#&gt; two 20 30 42</span></span></code></pre></div>
<p>One of R’s strengths is the ease of creating new functions. Objects in the function body remain local to the function, and any data type may be returned. Here is an example user-created function:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-1"></a> </span>
<span id="cb4-2"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-2"></a><span class="co"># Declare function “f” with parameters “x”, “y“</span></span>
<span id="cb4-3"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-3"></a><span class="co"># that returns a linear combination of x and y.</span></span>
<span id="cb4-4"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-4"></a>f &lt;-<span class="st"> </span><span class="cf">function</span>(x, y) {</span>
<span id="cb4-5"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-5"></a>  z &lt;-<span class="st"> </span><span class="dv">3</span> <span class="op">*</span><span class="st"> </span>x <span class="op">+</span><span class="st"> </span><span class="dv">4</span> <span class="op">*</span><span class="st"> </span>y</span>
<span id="cb4-6"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-6"></a>  <span class="kw">return</span>(z) <span class="co">## the return() function is optional here</span></span>
<span id="cb4-7"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-7"></a>}</span>
<span id="cb4-8"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-8"></a><span class="kw">f</span>(<span class="dv">1</span>, <span class="dv">2</span>)</span>
<span id="cb4-9"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-9"></a><span class="co">#&gt; [1] 11</span></span>
<span id="cb4-10"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-10"></a></span>
<span id="cb4-11"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-11"></a><span class="kw">f</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="kw">c</span>(<span class="dv">5</span>,<span class="dv">3</span>,<span class="dv">4</span>))</span>
<span id="cb4-12"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-12"></a><span class="co">#&gt; [1] 23 18 25</span></span>
<span id="cb4-13"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-13"></a></span>
<span id="cb4-14"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-14"></a><span class="kw">f</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">4</span>)</span>
<span id="cb4-15"><a href="r-rstudio-cloud-and-bioconductor.html#cb4-15"></a><span class="co">#&gt; [1] 19 22 25</span></span></code></pre></div>
</div>
<div id="install-bioconductor-and-bioconductor-packages" class="section level2">
<h2><span class="header-section-number">10.4</span> Install Bioconductor and Bioconductor Packages</h2>
<p>Install latest version of Bioconductor by entering the commands below (the lines starting with “#&gt;” are the outpunt produced by the commands listed at the top):</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-1"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))</span>
<span id="cb5-2"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-2"></a>    <span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb5-3"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-3"></a>BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="dt">version =</span> <span class="st">&quot;3.12&quot;</span>)</span>
<span id="cb5-4"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-4"></a><span class="co">#&gt; &#39;getOption(&quot;repos&quot;)&#39; replaces Bioconductor standard repositories, see</span></span>
<span id="cb5-5"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-5"></a><span class="co">#&gt; &#39;?repositories&#39; for details</span></span>
<span id="cb5-6"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-6"></a><span class="co">#&gt; </span></span>
<span id="cb5-7"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-7"></a><span class="co">#&gt; replacement repositories:</span></span>
<span id="cb5-8"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-8"></a><span class="co">#&gt;     CRAN: https://packagemanager.rstudio.com/all/__linux__/focal/latest</span></span>
<span id="cb5-9"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-9"></a><span class="co">#&gt; Bioconductor version 3.12 (BiocManager 1.30.14), R 4.0.3 (2020-10-10)</span></span>
<span id="cb5-10"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-10"></a><span class="co">#&gt; Installation paths not writeable, unable to update packages</span></span>
<span id="cb5-11"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-11"></a><span class="co">#&gt;   path: /usr/local/lib/R/library</span></span>
<span id="cb5-12"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-12"></a><span class="co">#&gt;   packages:</span></span>
<span id="cb5-13"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-13"></a><span class="co">#&gt;     boot, class, cluster, codetools, foreign, KernSmooth, lattice, MASS,</span></span>
<span id="cb5-14"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-14"></a><span class="co">#&gt;     Matrix, mgcv, nlme, nnet, spatial, survival</span></span>
<span id="cb5-15"><a href="r-rstudio-cloud-and-bioconductor.html#cb5-15"></a><span class="co">#&gt; Old packages: &#39;BiocManager&#39;, &#39;bslib&#39;, &#39;sass&#39;, &#39;systemfonts&#39;</span></span></code></pre></div>
<p>Next, install specific packages:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-1"></a>BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="kw">c</span>(<span class="st">&quot;sangerseqR&quot;</span>, <span class="st">&quot;Biostrings&quot;</span>))</span>
<span id="cb6-2"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-2"></a><span class="co">#&gt; &#39;getOption(&quot;repos&quot;)&#39; replaces Bioconductor standard repositories, see</span></span>
<span id="cb6-3"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-3"></a><span class="co">#&gt; &#39;?repositories&#39; for details</span></span>
<span id="cb6-4"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-4"></a><span class="co">#&gt; </span></span>
<span id="cb6-5"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-5"></a><span class="co">#&gt; replacement repositories:</span></span>
<span id="cb6-6"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-6"></a><span class="co">#&gt;     CRAN: https://packagemanager.rstudio.com/all/__linux__/focal/latest</span></span>
<span id="cb6-7"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-7"></a><span class="co">#&gt; Bioconductor version 3.12 (BiocManager 1.30.14), R 4.0.3 (2020-10-10)</span></span>
<span id="cb6-8"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-8"></a><span class="co">#&gt; Warning: package(s) not installed when version(s) same as current; use `force = TRUE` to</span></span>
<span id="cb6-9"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-9"></a><span class="co">#&gt;   re-install: &#39;sangerseqR&#39; &#39;Biostrings&#39;</span></span>
<span id="cb6-10"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-10"></a><span class="co">#&gt; Installation paths not writeable, unable to update packages</span></span>
<span id="cb6-11"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-11"></a><span class="co">#&gt;   path: /usr/local/lib/R/library</span></span>
<span id="cb6-12"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-12"></a><span class="co">#&gt;   packages:</span></span>
<span id="cb6-13"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-13"></a><span class="co">#&gt;     boot, class, cluster, codetools, foreign, KernSmooth, lattice, MASS,</span></span>
<span id="cb6-14"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-14"></a><span class="co">#&gt;     Matrix, mgcv, nlme, nnet, spatial, survival</span></span>
<span id="cb6-15"><a href="r-rstudio-cloud-and-bioconductor.html#cb6-15"></a><span class="co">#&gt; Old packages: &#39;BiocManager&#39;, &#39;bslib&#39;, &#39;sass&#39;, &#39;systemfonts&#39;</span></span></code></pre></div>
<p>Now, we are ready to read a AB1 sequence file, save the chromatogram (electropherogram) as PDF file and the sequence as a text file:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-1"></a><span class="co"># load packages</span></span>
<span id="cb7-2"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-2"></a><span class="kw">library</span>(sangerseqR)</span>
<span id="cb7-3"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-3"></a><span class="kw">library</span>(Biostrings)</span>
<span id="cb7-4"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-4"></a></span>
<span id="cb7-5"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-5"></a><span class="co"># read the ab1 sequence file</span></span>
<span id="cb7-6"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-6"></a>sequence &lt;-<span class="st"> </span><span class="kw">readsangerseq</span>(<span class="st">&quot;./data/78_4212_91_EQ-CHF3_4239_SQ_F10_035.ab1&quot;</span>)</span>
<span id="cb7-7"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-7"></a><span class="co"># save the chromatogram to a pdf file</span></span>
<span id="cb7-8"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-8"></a><span class="kw">chromatogram</span>(sequence, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dt">width =</span> <span class="dv">100</span>, <span class="dt">height =</span> <span class="dv">2</span>, <span class="dt">filename =</span> <span class="st">&quot;./data/sequence_chromatogram.pdf&quot;</span>)</span>
<span id="cb7-9"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-9"></a><span class="co">#&gt; Chromatogram saved to ./data/sequence_chromatogram.pdf in the current working directory</span></span>
<span id="cb7-10"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-10"></a><span class="co"># save the sequence as a text file</span></span>
<span id="cb7-11"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-11"></a><span class="kw">primarySeq</span>(sequence, <span class="dt">string =</span> <span class="ot">TRUE</span>)</span>
<span id="cb7-12"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-12"></a><span class="co">#&gt; [1] &quot;NNNNNNNNNNNNNNNNNNNNNNNNNNNTAANNNANNCTTCTTTTTTACATTTAACAAGATTTTTTNTACATGAATATAAAGAAAGGTCTAAAAATGAAAAATTGATTTATTCTTTTAATAAAAGAAATAAATTTGTCACATTTTCTTGGAATTTTTTTTATATTTTTGAATTGGAATTTTTTTTAACTTCTCTTTTGACTAGGTTTNTAAATAATTTAGTTCTATCCAACTTATTCGATCAAATTAATCTATTAGAAAAAATAAATAATAATAATAAAAGTCAATATTTTTTATCAGAAAAAAGGATCTATAACCAAAATTCCTGTATTCATTATGTACGGTATCAAAATCGTTGTATTATGGCTTCGGAAGGCTTTTATTTTCATGATACGAACTGGATATATTTTATATTGAATATTTGGCAATTTTTTATGCATTTATGGATTCAACCTTTCAGGTTCTCAACAAAGCATTTTCAAAAGCAGAGTTTTTTTTTTCTGGGTTATCAATTTGGCAGAGAATCCAAACTGCTAAAAGTTCGATCTATTTCACTTGATAAATCACCTACAATTTATTCACGTTTAAAAAAAAATCTTTTAAAAACACAAATTGTATATCCAATAGATTTTCTAGCAAAAGAGGGTTTTTGTGATATTTCAGGTTATCCTATTAGTAGATCGACTTGGACTACNTCGACNGATGAAGAANTTTTATTGAATTTTAATAAAATTTGNAAAAGCTTTTATTTTTATTATGGGGGGTTGATAAAAAAAGACATTTTATANNNAATCANNTATATCCTCCGATTTTCATGTGCAAAAACNCTANCTCGTAANCNNNNAANNNNNAANNNNNNA&quot;</span></span>
<span id="cb7-13"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-13"></a></span>
<span id="cb7-14"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-14"></a><span class="kw">makeBaseCalls</span>(sequence, <span class="dt">ratio =</span> <span class="fl">0.33</span>)</span>
<span id="cb7-15"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-15"></a><span class="co">#&gt; Number of datapoints: 10197</span></span>
<span id="cb7-16"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-16"></a><span class="co">#&gt; Number of basecalls: 855</span></span>
<span id="cb7-17"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-17"></a><span class="co">#&gt; </span></span>
<span id="cb7-18"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-18"></a><span class="co">#&gt; Primary Basecalls: GGCATTTTCGCAGTCATCGACACTAATAAAAGATGCTTCTTTTTTACATTTAACAAGATTTTTTGTACATGAATATAAAGAAAGGTCTAAAAATGAAAAATTGATTTATTCTTTTAATAAAAGAAATAAATTTGTCACATTTTCTTGGAATTTTTTTTATATTTTTGAATTGGAATTTTTTTTAACTTCTCTTTTGACTAGGTTTCTAAATAATTTAGTTCTATCCAACTTATTCGATCAAATTAATCTATTAGAAAAAATAAATAATAATAATAAAAGTCAATATTTTTTATCAGAAAAAAGGATCTATAACCAAAATTCCTGTATTCATTATGTACGGTATCAAAATCGTTGTATTATGGCTTCGGAAGGCTTTTATTTTCATGATACGAACTGGATATATTTTATATTGAATATTTGGCAATTTTTTATGCATTTATGGATTCAACCTTTCAGGTTCTCAACAAAGCATTTTCAAAAGCAGAGTTTTTTTTTTCTGGGTTATCAATTTGGCAGAGAATCCAAACTGCTAAAAGTTCGATCTATTTCACTTGATAAATCACCTACAATTTATTCACGTTTAAAAAAAAATCTTTTAAAAACACAAATTGTATATCCAATAGATTTTCTAGCAAAAGAGGGTTTTTGTGATATTTCAGGTTATCCTATTAGTAGATCGACTTGGACTACATCGACAGATGAAGAAATTTTATTGAATTTTAATAAAATTTGGAAAAGCTTTTATTTTTATTATGGGGGGTTGATAAAAAAAGACATTTTATATAGAATCAAATATATCCTCCGATTTTCATGTGCAAAAACACTAGCTCGTAAACATGAAAAAAAAATATAGCA</span></span>
<span id="cb7-19"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-19"></a><span class="co">#&gt; </span></span>
<span id="cb7-20"><a href="r-rstudio-cloud-and-bioconductor.html#cb7-20"></a><span class="co">#&gt; Secondary Basecalls: GGGATATTCGCACTCGTCGTCACTAATAAAAGATGCTTCTTTTTTACATTTAACAAGATTTTTTGTACATGAATATAAAGAAAGGTCTAAAAATGAAAAATTGATTTATTCTTTTAATAAAAGAAATAAATTTGTCACATTTTCTTGGAATTTTTTTTATATTTTTGAATTGGAATTTTTTTTAACTTCTCTTTTGACTAGGTTTATAAATAATTTAGTTCTATCCAACTTATTCGATCAAATTAATCTATTAGAAAAAATAAATAATAATAATAAAAGTCAATATTTTTTATCAGAAAAAAGGATCTATAACCAAAATTCCTGTATTCATTATGTACGGTATCAAAATCGTTGTATTATGGCTTCGGAAGGCTTTTATTTTCATGATACGAACTGGATATATTTTATATTGAATATTTGGCAATTTTTTATGCATTTATGGATTCAACCTTTCAGGTTCTCAACAAAGCATTTTCAAAAGCAGAGTTTTTTTTTTCTGGGTTATCAATTTGGCAGAGAATCCAAACTGCTAAAAGTTCGATCTATTTCACTTGATAAATCACCTACAATTTATTCACGTTTAAAAAAAAATCTTTTAAAAACACAAATTGTATATCCAATAGATTTTCTAGCAAAAAAGGGTTTTTGTGATATTTCAGGTTATCCTATTAATAGATCGACTTGGACTACCTCTACTGATGAAGAAATTTTATTGAATTTTAATAAAATTTGGAAAAGCTTTTATTTTTATTATGGGGGGTTGATAAAAAAAGACATTTTATATACAATCAAATATATCCCCCGATTTTCATGTGCAAAAACACTAYCCCGTAAACTATAAGTYCAAATAGAGCA</span></span></code></pre></div>
<p>The DNA sequence chromatogram obtained from sequencing the PCR products from the unknown organism. This chromatogram was produced using the free and open-source software programs <a href="https://www.r-project.org">R</a> and <a href="https://bioconductor.org">Bioconductor</a> package <a href="https://bioconductor.org/packages/release/bioc/html/sangerseqR.html">sangerseqR</a>. The DNA sequence chromatogram obtained from sequencing the PCR products from the unknown organism. This chromatogram was produced using the free and open-source software programs <a href="https://www.r-project.org">R</a> and <a href="https://bioconductor.org">Bioconductor</a> package <a href="https://bioconductor.org/packages/release/bioc/html/sangerseqR.html">sangerseqR</a>.</p>
<div class="figure" style="text-align: center"><span id="fig:chrom"></span>
<img src="data/sequence_chromatogram.pdf" alt="The DNA sequence chromatogram obtained from sequencing the PCR products from the unknown organism. This chromatogram was produced using the free and open-source software programs R and Bioconductor package sangerseqR." width="70%" />
<p class="caption">
Figure 7.1: The DNA sequence chromatogram obtained from sequencing the PCR products from the unknown organism. This chromatogram was produced using the free and open-source software programs <a href="https://www.r-project.org">R</a> and <a href="https://bioconductor.org">Bioconductor</a> package <a href="https://bioconductor.org/packages/release/bioc/html/sangerseqR.html">sangerseqR</a>.
</p>
</div>
</div>
<div id="blast-from-r" class="section level2">
<h2><span class="header-section-number">10.5</span> BLAST From R</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-1"></a><span class="co"># BLAST search in R</span></span>
<span id="cb8-2"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-2"></a></span>
<span id="cb8-3"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-3"></a><span class="co"># set working directory</span></span>
<span id="cb8-4"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-4"></a><span class="kw">setwd</span>(<span class="st">&quot;~/Dropbox/R/bioinformatics&quot;</span>)</span>
<span id="cb8-5"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-5"></a></span>
<span id="cb8-6"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-6"></a><span class="co"># function testing if Bioconductor packages are installed</span></span>
<span id="cb8-7"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-7"></a>installBiocManager &lt;-<span class="st"> </span><span class="cf">function</span>(x) {</span>
<span id="cb8-8"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-8"></a>    <span class="cf">if</span> (x <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(<span class="kw">installed.packages</span>()) <span class="op">==</span><span class="st"> </span><span class="ot">FALSE</span>) {</span>
<span id="cb8-9"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-9"></a>        <span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>)) </span>
<span id="cb8-10"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-10"></a>            <span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb8-11"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-11"></a>        BiocManager<span class="op">::</span><span class="kw">install</span>(x, <span class="dt">ask =</span> <span class="ot">FALSE</span>)</span>
<span id="cb8-12"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-12"></a>    } <span class="cf">else</span> {</span>
<span id="cb8-13"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-13"></a>        <span class="kw">paste</span>(x, <span class="st">&quot;is already installed!&quot;</span>)</span>
<span id="cb8-14"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-14"></a>    }</span>
<span id="cb8-15"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-15"></a>}</span>
<span id="cb8-16"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-16"></a></span>
<span id="cb8-17"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-17"></a><span class="co"># install necessary Bioconductor packages</span></span>
<span id="cb8-18"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-18"></a>required_packages &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="st">&quot;sangerseqR&quot;</span>, <span class="st">&quot;Biostrings&quot;</span>, <span class="st">&quot;annotate&quot;</span>, <span class="st">&quot;msa&quot;</span>, </span>
<span id="cb8-19"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-19"></a>    <span class="st">&quot;ggmsa&quot;</span>)</span>
<span id="cb8-20"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-20"></a><span class="kw">lapply</span>(required_packages, installBiocManager)</span>
<span id="cb8-21"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-21"></a><span class="co">#&gt; [[1]]</span></span>
<span id="cb8-22"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-22"></a><span class="co">#&gt; [1] &quot;BiocManager is already installed!&quot;</span></span>
<span id="cb8-23"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-23"></a><span class="co">#&gt; </span></span>
<span id="cb8-24"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-24"></a><span class="co">#&gt; [[2]]</span></span>
<span id="cb8-25"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-25"></a><span class="co">#&gt; [1] &quot;sangerseqR is already installed!&quot;</span></span>
<span id="cb8-26"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-26"></a><span class="co">#&gt; </span></span>
<span id="cb8-27"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-27"></a><span class="co">#&gt; [[3]]</span></span>
<span id="cb8-28"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-28"></a><span class="co">#&gt; [1] &quot;Biostrings is already installed!&quot;</span></span>
<span id="cb8-29"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-29"></a><span class="co">#&gt; </span></span>
<span id="cb8-30"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-30"></a><span class="co">#&gt; [[4]]</span></span>
<span id="cb8-31"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-31"></a><span class="co">#&gt; [1] &quot;annotate is already installed!&quot;</span></span>
<span id="cb8-32"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-32"></a><span class="co">#&gt; </span></span>
<span id="cb8-33"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-33"></a><span class="co">#&gt; [[5]]</span></span>
<span id="cb8-34"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-34"></a><span class="co">#&gt; [1] &quot;msa is already installed!&quot;</span></span>
<span id="cb8-35"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-35"></a><span class="co">#&gt; </span></span>
<span id="cb8-36"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-36"></a><span class="co">#&gt; [[6]]</span></span>
<span id="cb8-37"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-37"></a><span class="co">#&gt; [1] &quot;ggmsa is already installed!&quot;</span></span>
<span id="cb8-38"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-38"></a></span>
<span id="cb8-39"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-39"></a><span class="co"># load packages</span></span>
<span id="cb8-40"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-40"></a><span class="kw">library</span>(Biostrings)</span>
<span id="cb8-41"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-41"></a><span class="kw">library</span>(annotate)</span>
<span id="cb8-42"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-42"></a><span class="kw">library</span>(msa)</span>
<span id="cb8-43"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-43"></a><span class="kw">library</span>(ggmsa)</span>
<span id="cb8-44"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-44"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb8-45"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-45"></a><span class="co"># load the file that contains the query sequence for the BLAST</span></span>
<span id="cb8-46"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-46"></a></span>
<span id="cb8-47"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-47"></a>query &lt;-<span class="st"> </span><span class="kw">readDNAStringSet</span>(<span class="st">&quot;data/seq.fasta&quot;</span>)</span>
<span id="cb8-48"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-48"></a></span>
<span id="cb8-49"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-49"></a><span class="co"># BLAST the sequence blast_output &lt;- blastSequences(query, as=&#39;data.frame&#39;)</span></span>
<span id="cb8-50"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-50"></a></span>
<span id="cb8-51"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-51"></a><span class="co"># Inspect the result detail(blast_output) print(blast_output, show = &#39;complete&#39;)</span></span>
<span id="cb8-52"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-52"></a></span>
<span id="cb8-53"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-53"></a><span class="co"># save the returned matches as a DNA string set class fasta &lt;-</span></span>
<span id="cb8-54"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-54"></a><span class="co"># DNAStringSet(c(blast_output$Hsp_qseq[1], blast_output$Hsp_hseq[c(1,3,5,7,9)]))</span></span>
<span id="cb8-55"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-55"></a><span class="co"># add the names of the returned sequences names(fasta) &lt;- c(rep(&#39;query&#39;,1),</span></span>
<span id="cb8-56"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-56"></a><span class="co"># blast_output$Hit_id[c(1,3,5,7,9)])</span></span>
<span id="cb8-57"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-57"></a></span>
<span id="cb8-58"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-58"></a>fasta &lt;-<span class="st"> </span><span class="kw">readDNAStringSet</span>(<span class="st">&quot;Equisetum_blast.fasta&quot;</span>)</span>
<span id="cb8-59"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-59"></a></span>
<span id="cb8-60"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-60"></a><span class="co"># perform multiple sequence alignment</span></span>
<span id="cb8-61"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-61"></a>aln &lt;-<span class="st"> </span><span class="kw">msa</span>(fasta)</span>
<span id="cb8-62"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-62"></a><span class="co">#&gt; use default substitution matrix</span></span>
<span id="cb8-63"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-63"></a><span class="co"># look at the result</span></span>
<span id="cb8-64"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-64"></a><span class="kw">DNAStringSet</span>(aln)</span>
<span id="cb8-65"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-65"></a><span class="co">#&gt; DNAStringSet object of length 6:</span></span>
<span id="cb8-66"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-66"></a><span class="co">#&gt;     width seq                                               names               </span></span>
<span id="cb8-67"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-67"></a><span class="co">#&gt; [1]   618 AGTTCTATCCAACTTATTCGATC...TGTGCAAAAACACTAGCTCGTAA gi|281371607|gb|G...</span></span>
<span id="cb8-68"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-68"></a><span class="co">#&gt; [2]   618 AGTTCTATCCAACTTATTCGATC...TGTGCAAAAACACTAGCTCGTAA gi|37993690|gb|AY...</span></span>
<span id="cb8-69"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-69"></a><span class="co">#&gt; [3]   618 AGTTCTATCCAACTTATTCGATC...TGTGCAAAAACACTAGCTCGTAA gi|359751518|gb|J...</span></span>
<span id="cb8-70"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-70"></a><span class="co">#&gt; [4]   618 AGTTCTATCCAACTTATTCGATC...TGTGCAAAAACACTA-------- gi|1988363571|gb|...</span></span>
<span id="cb8-71"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-71"></a><span class="co">#&gt; [5]   618 AGTTCTATCCAACTTATTCGATC...TGTGCAAAAAC------------ gi|402715855|gb|J...</span></span>
<span id="cb8-72"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-72"></a><span class="co">#&gt; [6]   618 AGTTCTATCCAACTTATTCGATC...TGTGCAAAAACNCTANCTCGTAA query</span></span>
<span id="cb8-73"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-73"></a><span class="co"># write the alignment to a file writeXStringSet(fasta,</span></span>
<span id="cb8-74"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-74"></a><span class="co"># file=&#39;Equisetum_blast.fasta&#39;, format=&#39;fasta&#39;, width=80)</span></span>
<span id="cb8-75"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-75"></a></span>
<span id="cb8-76"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-76"></a><span class="co"># plot the alignment ggmsa(fasta, color = &#39;Chemistry_NT&#39;, font = &#39;DroidSansMono&#39;,</span></span>
<span id="cb8-77"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-77"></a><span class="co"># start = 1, end = 600, char_width = 0.5, seq_name = TRUE) + facet_msa(field =</span></span>
<span id="cb8-78"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-78"></a><span class="co"># 80) ggsave(&#39;Equisetum_alignment.pdf&#39;, dpi = &#39;print&#39;, width = 8.5, height = 11,</span></span>
<span id="cb8-79"><a href="r-rstudio-cloud-and-bioconductor.html#cb8-79"></a><span class="co"># units = &#39;in&#39;)</span></span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:alignment"></span>
<img src="Equisetum_alignment.pdf" alt="The sequence alingment." width="70%" />
<p class="caption">
Figure 7.10: The sequence alingment.
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="the-human-genome.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": false,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/nikolaussucher/bioinformatics/edit/master/bioconductor.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
